compose: TestComposeCompare failed

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11467134?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11467134?buildTab=artifacts#/) on release-23.1 @ [e6d0a5283b82a9d5485101bf3508c16b1597f085](https://github.com/cockroachdb/cockroach/commits/e6d0a5283b82a9d5485101bf3508c16b1597f085):


```
        test_1        |             + 				Int:              s"10000000000000000",
        test_1        |             - 				Exp:              0,
        test_1        |             + 				Exp:              -16,
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((4:::INT8::INT8::INT8 ^ 0:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS col88986
        test_1        |             	)
        test_1        |             		AS tab50457
        test_1        |             ORDER BY
        test_1        |             	col88986 ASC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((4::INT8::INT8::INT8 ^ 0::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS col88986
        test_1        |             	)
        test_1        |             		AS tab50457
        test_1        |             ORDER BY
        test_1        |             	col88986 ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (693.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30902

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11485029?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11485029?buildTab=artifacts#/) on release-23.1 @ [b151af8ea747b3dcd7aebfc1aa0c6d5ed2bc5708](https://github.com/cockroachdb/cockroach/commits/b151af8ea747b3dcd7aebfc1aa0c6d5ed2bc5708):


```
        test_1        |             + 				Int:              s"7290000000000000000",
        test_1        |             - 				Exp:              0,
        test_1        |             + 				Exp:              -16,
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || power(9:::DECIMAL::DECIMAL::DECIMAL, 3:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col439904
        test_1        |             	)
        test_1        |             		AS "%q😥Ta\\uEFDDb͢239838"
        test_1        |             ORDER BY
        test_1        |             	col439904 DESC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || power(9::DECIMAL::DECIMAL::DECIMAL, 3::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col439904
        test_1        |             	)
        test_1        |             		AS "%q😥Ta\\uEFDDb͢239838"
        test_1        |             ORDER BY
        test_1        |             	col439904 DESC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1066.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11508357?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11508357?buildTab=artifacts#/) on release-23.1 @ [e9a99ca5edea2d3f838467323355be76d9d3ec92](https://github.com/cockroachdb/cockroach/commits/e9a99ca5edea2d3f838467323355be76d9d3ec92):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (242.41s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3888100230/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000001000000050000000000000000001840000000000000F0BF00000000000018400000000000002240FDD54F962789EE3F0000000000002240FDD54F962789EE3F000000000000F0BF0000000000001840000000000000F0BF"),
        test_1        |             + 	string("010300000000000000"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(6.0::FLOAT8::FLOAT8::FLOAT8, (4.0::FLOAT8::FLOAT8::FLOAT8 - 5.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, log(9.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col372921
        test_1        |             	)
        test_1        |             		AS tab189820
        test_1        |             ORDER BY
        test_1        |             	col372921 ASC NULLS LAST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(6.0:::FLOAT8::FLOAT8::FLOAT8, (4.0:::FLOAT8::FLOAT8::FLOAT8 - 5.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, log(9.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col372921
        test_1        |             	)
        test_1        |             		AS tab189820
        test_1        |             ORDER BY
        test_1        |             	col372921 ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (870.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11516843?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11516843?buildTab=artifacts#/) on release-23.1 @ [e9a99ca5edea2d3f838467323355be76d9d3ec92](https://github.com/cockroachdb/cockroach/commits/e9a99ca5edea2d3f838467323355be76d9d3ec92):


```
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	[]uint8{
        test_1        |               		0x21,
        test_1        |               		0x08,
        test_1        |             - 		0x1a,
        test_1        |             + 		0x02,
        test_1        |               		0x28,
        test_1        |               		0x50,
        test_1        |               		0x08,
        test_1        |               	},
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepointm(2.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 1::INT8::INT4::INT4, 6::INT8::INT4::INT4)::BYTEA
        test_1        |             				AS "%d2col315058"
        test_1        |             	)
        test_1        |             		AS tab212238
        test_1        |             ORDER BY
        test_1        |             	"%d2col315058" DESC;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepointm(2.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 1:::INT8::INT4::INT4, 6:::INT8::INT4::INT4)::BYTES
        test_1        |             				AS "%d2col315058"
        test_1        |             	)
        test_1        |             		AS tab212238
        test_1        |             ORDER BY
        test_1        |             	"%d2col315058" DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (916.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11532664?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11532664?buildTab=artifacts#/) on release-23.1 @ [b05ca5cee319a3c0778fb5e50309e5f30507cd2b](https://github.com/cockroachdb/cockroach/commits/b05ca5cee319a3c0778fb5e50309e5f30507cd2b):


```
        cockroach1_1  | status:              initialized new cluster
        cockroach1_1  | nodeID:              1
        postgres_1    | CREATE EXTENSION
        postgres_1    | Loading PostGIS extensions into postgres
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-08-29 05:17:27.559 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2023-08-29 05:17:27.561 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-08-29 05:17:27.563 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-08-29 05:17:27.565 UTC [51] LOG:  shutting down
        postgres_1    | 2023-08-29 05:17:27.658 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-08-29 05:17:27.680 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-08-29 05:17:27.681 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-08-29 05:17:27.681 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-08-29 05:17:27.683 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-08-29 05:17:27.688 UTC [68] LOG:  database system was shut down at 2023-08-29 05:17:27 UTC
        postgres_1    | 2023-08-29 05:17:27.693 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-08-29 05:17:29.916 UTC [78] ERROR:  foreign key constraint "t͞able\\u62FD3_col3_6_co😾l3_0_fkey" cannot be implemented
        postgres_1    | 2023-08-29 05:17:29.916 UTC [78] DETAIL:  Key columns "co😾l3_0" and "col�1_0" are of incompatible types: "char" and name.
        postgres_1    | 2023-08-29 05:17:29.916 UTC [78] STATEMENT:  ALTER TABLE "t͞able\\u62FD3" ADD FOREIGN KEY (col3_6, co😾l3_0) REFERENCES "%pta'ble1" (col1_8, "col�1_0")
        test_1        | --- FAIL: TestCompare (606.47s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (1.39s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: ALTER TABLE "t͞able\\u62FD3" ADD FOREIGN KEY (col3_6, co😾l3_0) REFERENCES "%pta'ble1" (col1_8, "col�1_0")
        test_1        |         compare_test.go:173: postgres: exec: ERROR: foreign key constraint "t͞able\\u62FD3_col3_6_co😾l3_0_fkey" cannot be implemented (SQLSTATE 42804)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11606170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11606170?buildTab=artifacts#/) on release-23.1 @ [58ac87a9b24e1cfb446143dafe04808ea186637f](https://github.com/cockroachdb/cockroach/commits/58ac87a9b24e1cfb446143dafe04808ea186637f):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (229.32s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare89666344/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	string("0"),
        test_1        |             - 	nil,
        test_1        |             + 	string("01010000000000000000B063C00000000000E05040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			bpchar('00:04:00':::TIME::TIME::TIME)::CHAR AS col271409,
        test_1        |             			st_pointfromgeohash('B':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS col271410
        test_1        |             	)
        test_1        |             		AS tab181043
        test_1        |             ORDER BY
        test_1        |             	col271409 ASC, col271410 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			bpchar('00:04:00'::TIME::TIME::TIME)::CHAR AS col271409,
        test_1        |             			st_pointfromgeohash('B'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS col271410
        test_1        |             	)
        test_1        |             		AS tab181043
        test_1        |             ORDER BY
        test_1        |             	col271409 ASC, col271410 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (857.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11616149?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11616149?buildTab=artifacts#/) on release-23.1 @ [fde98bbb2423906a7e3bfd9cd7956bc3f29918ad](https://github.com/cockroachdb/cockroach/commits/fde98bbb2423906a7e3bfd9cd7956bc3f29918ad):


```
        postgres_1    | 			WHERE
        postgres_1    | 				NULL::BOOL
        postgres_1    | 			ORDER BY
        postgres_1    | 				"t\\u5446ab59923"."c,ol3_̀6" ASC NULLS LAST
        postgres_1    | 		)
        postgres_1    | 			AS tab59925
        postgres_1    | 	ORDER BY
        postgres_1    | 		"c ol114126" DESC NULLS LAST
        test_1        | --- FAIL: TestCompare (696.56s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (91.42s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2604141028/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(0),
        test_1        |             + 	int16(1),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(tanh(2.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "col}114206") AS tab59991
        test_1        |             ORDER BY
        test_1        |             	"col}114206" ASC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(tanh(2.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "col}114206") AS tab59991
        test_1        |             ORDER BY
        test_1        |             	"col}114206" ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (720.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.10-rc]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11645262?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11645262?buildTab=artifacts#/) on release-23.1 @ [6db39099511584d8b41094237f5b26333fb4c7be](https://github.com/cockroachdb/cockroach/commits/6db39099511584d8b41094237f5b26333fb4c7be):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 830657 [running]:
        test_1        | testing.tRunner.func1.2({0x22a4680, 0xc002814640})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22a4680, 0xc002814640})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45b13a0?, {0x31c4cc8?, 0xc0027987b0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x3188760?, 0x45b13a0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc000c5c300)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc001e59b90, {0xc0045a4230?, 0x5, 0xc001e59b90?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc000096968?, 0x24a728b?, {0xc0045a4230?, 0x20df9a0?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20df9a0?, {0x24a728b, 0x5}, 0x5, 0x0?, {0xc00134fa48, 0x3, 0xc0009b6690?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc00296d1e0?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x26831a0?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc00296d1e0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc00296d1e0, 0xc001e59b30)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (630.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.10-rc]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11709013?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11709013?buildTab=artifacts#/) on release-23.1 @ [50c96a6208c5a211a6174654ac3dd6487a046006](https://github.com/cockroachdb/cockroach/commits/50c96a6208c5a211a6174654ac3dd6487a046006):


```
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-09-10 05:17:49.079 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-09-10 05:17:49.079 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-09-10 05:17:49.079 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-09-10 05:17:49.082 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-09-10 05:17:49.087 UTC [67] LOG:  database system was shut down at 2023-09-10 05:17:49 UTC
        postgres_1    | 2023-09-10 05:17:49.091 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-09-10 05:17:50.495 UTC [77] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-09-10 05:17:50.495 UTC [77] STATEMENT:  CREATE TABLE table5 ("co\\x51l5_0" INT4 NOT NULL, col5_1 BIT(38) NOT NULL, col5_2 CHAR NOT NULL, col5̽_3 TEXT NOT NULL, col5_4 REGTYPE NOT NULL, col5_5 BYTEA NOT NULL, col5_6 CHAR, "co\gl5_7" REGTYPE, "col'5_8" INT8 NOT NULL, col5_9 TEXT NOT NULL, col5_10 UUID NOT NULL, c̅ol5_11 VARCHAR NOT NULL, "cͮol5_\\xaa12" DATE, "co l5_13" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col5_5 IS NULL THEN e'\x00'::TEXT ELSE ']'::TEXT END) STORED, co̽l5_14 INT8 NOT NULL GENERATED ALWAYS AS ("co\\x51l5_0" + "col'5_8") STORED, col5_15 INT8 NOT NULL GENERATED ALWAYS AS ("col'5_8" + (-5602317532961290773)::INT8) STORED, "'col5_16" TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_4 IS NULL THEN e'[ o;Cu\f'::TEXT ELSE e'\U00002603'::TEXT END) STORED, "c%qo̯l5_%q17" INT8 NOT NULL GENERATED ALWAYS AS ("co\\x51l5_0" + "col'5_8") STORED, PRIMARY KEY (col5_9, col5_4, col5_10, c̅ol5_11, "co\\x51l5_0", col5_5, "col'5_8", col5_15));
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 ("'col5_16", ("co\\x51l5_0" + "col'5_8"), col5̽_3 ASC, col5_4 DESC, col5_9 ASC) INCLUDE (col5_1, "co l5_13");
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 (c̅ol5_11 ASC, col5_10 ASC, col5_2, col5_4 DESC, col5̽_3 ASC, "'col5_16", col5_15 DESC, co̽l5_14 ASC, "c%qo̯l5_%q17" ASC, ("co\\x51l5_0" + (-1560331512)::INT8) ASC, col5_1 ASC, col5_9, col5_6 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 (col5_5 DESC, col5_10, "cͮol5_\\xaa12", col5_9, "col'5_8", "'col5_16", "c%qo̯l5_%q17" DESC, col5_4, col5̽_3 DESC, c̅ol5_11 DESC, "co l5_13" DESC, col5_15 ASC, lower(col5_2));
        postgres_1    | 	CREATE INDEX ON table5 (col5̽_3 ASC, col5_6 DESC, ("co\\x51l5_0" + "col'5_8") ASC, col5_1 DESC, "'col5_16" ASC, "col'5_8" ASC, col5_9 ASC, "co\gl5_7" ASC, lower(CAST(col5_10 AS TEXT)) ASC, "cͮol5_\\xaa12", col5_15 DESC, col5_10 ASC, lower(CAST(col5_1 AS TEXT)) ASC, c̅ol5_11 DESC, co̽l5_14 DESC, "c%qo̯l5_%q17", col5_5 ASC) INCLUDE (col5_2);
        postgres_1    | 	CREATE INDEX ON table5 (co̽l5_14 ASC, col5_10 DESC, col5_5 ASC, "'col5_16", col5_9 ASC, col5̽_3, col5_4 ASC, col5_1 ASC, "cͮol5_\\xaa12", "c%qo̯l5_%q17" ASC, lower(CAST(col5_1 AS TEXT)) ASC, "col'5_8" DESC, "co l5_13", c̅ol5_11 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 ("co\\x51l5_0", col5_10);
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 ("co\\x51l5_0");
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.35s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.33s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table5 ("co\\x51l5_0" INT4 NOT NULL, col5_1 BIT(38) NOT NULL, col5_2 CHAR NOT NULL, col5̽_3 TEXT NOT NULL, col5_4 REGTYPE NOT NULL, col5_5 BYTEA NOT NULL, col5_6 CHAR, "co\gl5_7" REGTYPE, "col'5_8" INT8 NOT NULL, col5_9 TEXT NOT NULL, col5_10 UUID NOT NULL, c̅ol5_11 VARCHAR NOT NULL, "cͮol5_\\xaa12" DATE, "co l5_13" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col5_5 IS NULL THEN e'\x00'::TEXT ELSE ']'::TEXT END) STORED, co̽l5_14 INT8 NOT NULL GENERATED ALWAYS AS ("co\\x51l5_0" + "col'5_8") STORED, col5_15 INT8 NOT NULL GENERATED ALWAYS AS ("col'5_8" + (-5602317532961290773)::INT8) STORED, "'col5_16" TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_4 IS NULL THEN e'[ o;Cu\f'::TEXT ELSE e'\U00002603'::TEXT END) STORED, "c%qo̯l5_%q17" INT8 NOT NULL GENERATED ALWAYS AS ("co\\x51l5_0" + "col'5_8") STORED, PRIMARY KEY (col5_9, col5_4, col5_10, c̅ol5_11, "co\\x51l5_0", col5_5, "col'5_8", col5_15));
        test_1        |             CREATE UNIQUE INDEX ON table5 ("'col5_16", ("co\\x51l5_0" + "col'5_8"), col5̽_3 ASC, col5_4 DESC, col5_9 ASC) INCLUDE (col5_1, "co l5_13");
        test_1        |             CREATE UNIQUE INDEX ON table5 (c̅ol5_11 ASC, col5_10 ASC, col5_2, col5_4 DESC, col5̽_3 ASC, "'col5_16", col5_15 DESC, co̽l5_14 ASC, "c%qo̯l5_%q17" ASC, ("co\\x51l5_0" + (-1560331512)::INT8) ASC, col5_1 ASC, col5_9, col5_6 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_5 DESC, col5_10, "cͮol5_\\xaa12", col5_9, "col'5_8", "'col5_16", "c%qo̯l5_%q17" DESC, col5_4, col5̽_3 DESC, c̅ol5_11 DESC, "co l5_13" DESC, col5_15 ASC, lower(col5_2));
        test_1        |             CREATE INDEX ON table5 (col5̽_3 ASC, col5_6 DESC, ("co\\x51l5_0" + "col'5_8") ASC, col5_1 DESC, "'col5_16" ASC, "col'5_8" ASC, col5_9 ASC, "co\gl5_7" ASC, lower(CAST(col5_10 AS TEXT)) ASC, "cͮol5_\\xaa12", col5_15 DESC, col5_10 ASC, lower(CAST(col5_1 AS TEXT)) ASC, c̅ol5_11 DESC, co̽l5_14 DESC, "c%qo̯l5_%q17", col5_5 ASC) INCLUDE (col5_2);
        test_1        |             CREATE INDEX ON table5 (co̽l5_14 ASC, col5_10 DESC, col5_5 ASC, "'col5_16", col5_9 ASC, col5̽_3, col5_4 ASC, col5_1 ASC, "cͮol5_\\xaa12", "c%qo̯l5_%q17" ASC, lower(CAST(col5_1 AS TEXT)) ASC, "col'5_8" DESC, "co l5_13", c̅ol5_11 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table5 ("co\\x51l5_0", col5_10);
        test_1        |             CREATE UNIQUE INDEX ON table5 ("co\\x51l5_0");
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.10-rc]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11734767?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11734767?buildTab=artifacts#/) on release-23.1 @ [a3573e6b39530731621f90717649feac2c5dd47e](https://github.com/cockroachdb/cockroach/commits/a3573e6b39530731621f90717649feac2c5dd47e):


```
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab326794.tableoid ASC NULLS LAST
        postgres_1    | 		)
        postgres_1    | 			AS tab326796
        postgres_1    | 	ORDER BY
        postgres_1    | 		"col,590707" DESC, "c!o%89l590708" ASC NULLS FIRST
        test_1        | --- FAIL: TestCompare (1094.53s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (489.07s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare277502422/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int32(0),
        test_1        |             + 	int32(1),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4((5.0:::FLOAT8::FLOAT8::FLOAT8 / 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col590766)
        test_1        |             		AS "tab�326822"
        test_1        |             ORDER BY
        test_1        |             	col590766 ASC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4((5.0::FLOAT8::FLOAT8::FLOAT8 / 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col590766)
        test_1        |             		AS "tab�326822"
        test_1        |             ORDER BY
        test_1        |             	col590766 ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1118.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.10-rc]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11786704?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11786704?buildTab=artifacts#/) on release-23.1 @ [3224fa603020b8226d4ecbe4d520a2651b6b97d5](https://github.com/cockroachdb/cockroach/commits/3224fa603020b8226d4ecbe4d520a2651b6b97d5):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (409.33s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2659823521/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000002240000000000000144000000000000022400000000000001C4000000000000010400000000000001C400000000000001040000000000000144000000000000022400000000000001440"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(9.0:::FLOAT8::FLOAT8::FLOAT8, 5.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8::FLOAT8, 0:::INT8::INT4::INT4)::GEOMETRY
        test_1        |             				AS col542405
        test_1        |             	)
        test_1        |             		AS tab308231
        test_1        |             ORDER BY
        test_1        |             	col542405 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(9.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8::FLOAT8, 0::INT8::INT4::INT4)::GEOMETRY
        test_1        |             				AS col542405
        test_1        |             	)
        test_1        |             		AS tab308231
        test_1        |             ORDER BY
        test_1        |             	col542405 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1039.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.11-rc]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11801620?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11801620?buildTab=artifacts#/) on release-23.1 @ [12a0fdf76785787a3a7e83198f1adfd7184ea910](https://github.com/cockroachdb/cockroach/commits/12a0fdf76785787a3a7e83198f1adfd7184ea910):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (479.77s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare332584496/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000002240AA4C58E87AB6FB3F000000000000224000000000000014400000000000001C4000000000000014400000000000001C40AA4C58E87AB6FB3F0000000000002240AA4C58E87AB6FB3F"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(9.0::FLOAT8::FLOAT8::FLOAT8, sqrt(3.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col885471
        test_1        |             	)
        test_1        |             		AS "%f9tab457850"
        test_1        |             ORDER BY
        test_1        |             	col885471 ASC NULLS LAST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(9.0:::FLOAT8::FLOAT8::FLOAT8, sqrt(3.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8::FLOAT8, 5.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col885471
        test_1        |             	)
        test_1        |             		AS "%f9tab457850"
        test_1        |             ORDER BY
        test_1        |             	col885471 ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1107.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11814664?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11814664?buildTab=artifacts#/) on release-23.1 @ [12a0fdf76785787a3a7e83198f1adfd7184ea910](https://github.com/cockroachdb/cockroach/commits/12a0fdf76785787a3a7e83198f1adfd7184ea910):


```
        test_1        |               				Int:              s"0",
        test_1        |             - 				Exp:              0,
        test_1        |             + 				Exp:              -20,
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(CASE WHEN false::BOOL THEN (6:::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[] ELSE NULL::DECIMAL[] END::DECIMAL[] || (0:::DECIMAL::DECIMAL::DECIMAL / 5:::INT8::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS "%qcol437327"
        test_1        |             	)
        test_1        |             		AS tab274531
        test_1        |             ORDER BY
        test_1        |             	"%qcol437327";
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(CASE WHEN false::BOOL THEN (6::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[] ELSE NULL::DECIMAL[] END::DECIMAL[] || (0::DECIMAL::DECIMAL::DECIMAL / 5::INT8::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS "%qcol437327"
        test_1        |             	)
        test_1        |             		AS tab274531
        test_1        |             ORDER BY
        test_1        |             	"%qcol437327";
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1081.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11864946?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11864946?buildTab=artifacts#/) on release-23.1 @ [ef0a46c415ad61024743b3724e0243c8d9188cbb](https://github.com/cockroachdb/cockroach/commits/ef0a46c415ad61024743b3724e0243c8d9188cbb):


```
        test_1        | --- FAIL: TestCompare (989.18s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (384.80s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3456258790/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("1"),
        test_1        |             + 	string("1.0000000000000000"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			text((1:::DECIMAL::DECIMAL::DECIMAL ^ 1:::INT8::INT8::INT8)::DECIMAL::DECIMAL)::STRING AS "\\u78CCcol526444"
        test_1        |             	)
        test_1        |             		AS "'tab309625"
        test_1        |             ORDER BY
        test_1        |             	"\\u78CCcol526444";
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			text((1::DECIMAL::DECIMAL::DECIMAL ^ 1::INT8::INT8::INT8)::DECIMAL::DECIMAL)::TEXT AS "\\u78CCcol526444"
        test_1        |             	)
        test_1        |             		AS "'tab309625"
        test_1        |             ORDER BY
        test_1        |             	"\\u78CCcol526444";
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1013.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.9-rc.FROZEN]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11896210?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11896210?buildTab=artifacts#/) on release-23.1 @ [ffb4d004abe8c60fe256255550fe980d41f4e170](https://github.com/cockroachdb/cockroach/commits/ffb4d004abe8c60fe256255550fe980d41f4e170):


```
        test_1        |     --- FAIL: TestCompare/postgres (281.70s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2354476104/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("BOX(3 0,5 8)"),
        test_1        |             + 	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_combinebbox(NULL::BOX2D::BOX2D, st_makeenvelope(3.0:::FLOAT8::FLOAT8::FLOAT8, 8.0:::FLOAT8::FLOAT8::FLOAT8, 5.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::BOX2D
        test_1        |             				AS col382164
        test_1        |             	)
        test_1        |             		AS "t\\u661Ba'b
        test_1        |             222982"
        test_1        |             ORDER BY
        test_1        |             	CAST(col382164 AS STRING) DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_combinebbox(NULL::BOX2D::BOX2D, st_makeenvelope(3.0::FLOAT8::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::BOX2D
        test_1        |             				AS col382164
        test_1        |             	)
        test_1        |             		AS "t\\u661Ba'b
        test_1        |             222982"
        test_1        |             ORDER BY
        test_1        |             	CAST(col382164 AS TEXT) DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (910.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.11-rc]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11914247?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11914247?buildTab=artifacts#/) on release-23.1 @ [b08ec859debd7530b48f71405a700961596b3563](https://github.com/cockroachdb/cockroach/commits/b08ec859debd7530b48f71405a700961596b3563):


```
        postgres_1    | 		col688669 NULLS FIRST,
        postgres_1    | 		col688670 DESC NULLS FIRST,
        postgres_1    | 		col688671 NULLS LAST,
        postgres_1    | 		col688672 DESC NULLS FIRST,
        postgres_1    | 		"col%q688673",
        postgres_1    | 		"C""ol688674" ASC NULLS FIRST
        test_1        | --- FAIL: TestCompare (1101.19s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (496.70s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3884827821/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("G0.863691154450617"),
        test_1        |             + 	string("G0.8636911544506167"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT ('G':::STRING::STRING::STRING || cot(4.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::STRING AS col688687)
        test_1        |             		AS "tab\\x24414855"
        test_1        |             ORDER BY
        test_1        |             	col688687 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT ('G'::TEXT::TEXT::TEXT || cot(4.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::TEXT AS col688687)
        test_1        |             		AS "tab\\x24414855"
        test_1        |             ORDER BY
        test_1        |             	col688687 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1125.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.11-rc]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11930876?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11930876?buildTab=artifacts#/) on release-23.1 @ [b08ec859debd7530b48f71405a700961596b3563](https://github.com/cockroachdb/cockroach/commits/b08ec859debd7530b48f71405a700961596b3563):


```
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			B'0111'::BIT(26) AS "c%3eol830062", tab459888.co̵l830061 AS col830063
        test_1        |             		FROM
        test_1        |             			(
        test_1        |             				VALUES
        test_1        |             					(B'0011'::BIT(26), NULL::GEOMETRY),
        test_1        |             					(B'0101'::BIT(26), NULL::GEOMETRY),
        test_1        |             					(
        test_1        |             						B'0000'::BIT(26),
        test_1        |             						st_makeenvelope(4.0::FLOAT8::FLOAT8::FLOAT8, (5.0::FLOAT8::FLOAT8::FLOAT8 * 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             					)
        test_1        |             			)
        test_1        |             				AS tab459888 (col830060, co̵l830061)
        test_1        |             		ORDER BY
        test_1        |             			tab459888.co̵l830061 NULLS FIRST
        test_1        |             	)
        test_1        |             		AS tab459889
        test_1        |             ORDER BY
        test_1        |             	"c%3eol830062" ASC, col830063 ASC NULLS FIRST;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (37.62s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: cockroach2
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare678405014/002/mutators.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach2 to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("127"),
        test_1        |             + 	string(``e'e\'tabl\\U0001F602e5\''``),
        test_1        |               }))
        test_1        |             cockroach2:
        test_1        |             SELECT * FROM (SELECT regclass(127:::INT8::INT8)::REGCLASS AS col35184) AS tab22083 ORDER BY col35184 ASC NULLS LAST;
        test_1        |             
        test_1        |             cockroach1: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (639.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.11-rc]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11963984?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11963984?buildTab=artifacts#/) on release-23.1 @ [83c13d4a3847b9341e5d00a4936fdc99cbf27b57](https://github.com/cockroachdb/cockroach/commits/83c13d4a3847b9341e5d00a4936fdc99cbf27b57):


```
        test_1        |             ORDER BY
        test_1        |             	col65787;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 116485 [running]:
        test_1        | testing.tRunner.func1.2({0x22a89c0, 0xc001623c80})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22a89c0, 0xc001623c80})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45c53a0?, {0x31d7a48?, 0xc0020b0e10?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x319b520?, 0x45c53a0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc0002cf680)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc00127df50, {0xc00127df80?, 0x3, 0xc00127df50?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc001a81968?, 0x24ab82b?, {0xc00127df80?, 0x20e3840?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20e3840?, {0x24ab82b, 0x5}, 0x3, 0x0?, {0xc001677a48, 0x3, 0xc000e8ca50?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc00197bba0?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x2687cb8?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc00197bba0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc00197bba0, 0xc00127def0)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (98.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11994304?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/11994304?buildTab=artifacts#/) on release-23.1 @ [0deb53feff42298da826b13a0370d81eb9203e99](https://github.com/cockroachdb/cockroach/commits/0deb53feff42298da826b13a0370d81eb9203e99):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-09-30 05:17:02.388 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2023-09-30 05:17:02.390 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-09-30 05:17:02.391 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-09-30 05:17:02.394 UTC [51] LOG:  shutting down
        postgres_1    | 2023-09-30 05:17:02.489 UTC [49] LOG:  database system is shut down
        test_1        | I230930 05:17:02.536747 24 testutils/soon.go:41  [-] 1  SucceedsSoon: dial tcp 172.18.0.3:5432: connect: connection refused
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-09-30 05:17:02.611 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-09-30 05:17:02.612 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-09-30 05:17:02.612 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-09-30 05:17:02.615 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-09-30 05:17:02.620 UTC [68] LOG:  database system was shut down at 2023-09-30 05:17:02 UTC
        postgres_1    | 2023-09-30 05:17:02.625 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-09-30 05:17:04.639 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-09-30 05:17:04.639 UTC [78] STATEMENT:  CREATE TABLE "tab{\\U00106225le2" (col2_0 TSVECTOR NOT NULL, col2_1 TEXT NOT NULL, col2_2 "char" NOT NULL, "col�2\\x89_%bc3" TIMESTAMP NOT NULL, "co!l2_4" INTERVAL NOT NULL, "col 2_""5" FLOAT4 NOT NULL, cő😟l2_6 DATE NULL, col2_7 DECIMAL[] NOT NULL, col2_8 BIT(15) NOT NULL, col2͍_9 TEXT GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "co!l2_4" IS NULL THEN e'\x15E~'::TEXT ELSE e'\x11Z'::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col�2\\x89_%bc3" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) STORED, col2_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col2_2)) STORED, col2_14 TEXT GENERATED ALWAYS AS (lower(CAST(col2_8 AS TEXT))) STORED, col2_15 FLOAT8 NOT NULL GENERATED ALWAYS AS ("col 2_""5" + (-0.2320139855146408)::FLOAT8) STORED, "co
        postgres_1    | 	l2_16" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_0 AS TEXT))) STORED, PRIMARY KEY ("co!l2_4", "col�2\\x89_%bc3", col2_1, col2_7, col2_10, col2_8, col2_11, col2_12, col2_2, "col 2_""5"));
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.44s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.09s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "tab{\\U00106225le2" (col2_0 TSVECTOR NOT NULL, col2_1 TEXT NOT NULL, col2_2 "char" NOT NULL, "col�2\\x89_%bc3" TIMESTAMP NOT NULL, "co!l2_4" INTERVAL NOT NULL, "col 2_""5" FLOAT4 NOT NULL, cő😟l2_6 DATE NULL, col2_7 DECIMAL[] NOT NULL, col2_8 BIT(15) NOT NULL, col2͍_9 TEXT GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "co!l2_4" IS NULL THEN e'\x15E~'::TEXT ELSE e'\x11Z'::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col�2\\x89_%bc3" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) STORED, col2_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col2_2)) STORED, col2_14 TEXT GENERATED ALWAYS AS (lower(CAST(col2_8 AS TEXT))) STORED, col2_15 FLOAT8 NOT NULL GENERATED ALWAYS AS ("col 2_""5" + (-0.2320139855146408)::FLOAT8) STORED, "co
        test_1        |             l2_16" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_0 AS TEXT))) STORED, PRIMARY KEY ("co!l2_4", "col�2\\x89_%bc3", col2_1, col2_7, col2_10, col2_8, col2_11, col2_12, col2_2, "col 2_""5"));
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12001511?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12001511?buildTab=artifacts#/) on release-23.1 @ [0deb53feff42298da826b13a0370d81eb9203e99](https://github.com/cockroachdb/cockroach/commits/0deb53feff42298da826b13a0370d81eb9203e99):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1625023031/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000000840000000000000184000000000000008400000000000000840000000000000000000000000000008400000000000000000000000000000184000000000000008400000000000001840"),
        test_1        |               	s"2000-01-07 00:00:00 +0000 UTC",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col255767,
        test_1        |             			'2000-01-07':::DATE::DATE AS col255768
        test_1        |             	)
        test_1        |             		AS tab147536
        test_1        |             ORDER BY
        test_1        |             	col255767 DESC NULLS FIRST, col255768 DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col255767,
        test_1        |             			'2000-01-07'::DATE::DATE AS col255768
        test_1        |             	)
        test_1        |             		AS tab147536
        test_1        |             ORDER BY
        test_1        |             	col255767 DESC NULLS FIRST, col255768 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (843.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12042313?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12042313?buildTab=artifacts#/) on release-23.1 @ [1bf58a144bf53b0480cdd108d269fc7724875cf1](https://github.com/cockroachdb/cockroach/commits/1bf58a144bf53b0480cdd108d269fc7724875cf1):


```
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2030486832/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("C0.69314718055994530942"),
        test_1        |             + 	string("C0.6931471805599453"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			COALESCE(
        test_1        |             				st_ashexewkb(NULL::GEOMETRY::GEOMETRY)::STRING,
        test_1        |             				('C':::STRING::STRING::STRING || ln(2:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::STRING
        test_1        |             			)
        test_1        |             				AS col846563
        test_1        |             	)
        test_1        |             		AS tab507002
        test_1        |             ORDER BY
        test_1        |             	col846563;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			COALESCE(
        test_1        |             				st_ashexewkb(NULL::GEOMETRY::GEOMETRY)::TEXT,
        test_1        |             				('C'::TEXT::TEXT::TEXT || ln(2::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::TEXT
        test_1        |             			)
        test_1        |             				AS col846563
        test_1        |             	)
        test_1        |             		AS tab507002
        test_1        |             ORDER BY
        test_1        |             	col846563;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1123.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12074359?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12074359?buildTab=artifacts#/) on release-23.1 @ [85b8891c39752799b28e0b8eb9e805422ab5b427](https://github.com/cockroachdb/cockroach/commits/85b8891c39752799b28e0b8eb9e805422ab5b427):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2039952195/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("01010000000000000000A976400000000000706440"),
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_affine(st_pointfromgeohash('G':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY::GEOMETRY, 8.0:::FLOAT8::FLOAT8::FLOAT8, (((7.0:::FLOAT8::FLOAT8::FLOAT8 * 1.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 * 9.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 / 8.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 8.0:::FLOAT8::FLOAT8::FLOAT8, 5.0:::FLOAT8::FLOAT8::FLOAT8, (8.0:::FLOAT8::FLOAT8::FLOAT8 + 3.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, (COALESCE(1.0:::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8)::FLOAT8 * 6.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "col
309879",
        test_1        |             			NULL::TIME[] AS col309880
        test_1        |             	)
        test_1        |             		AS tab170183
        test_1        |             ORDER BY
        test_1        |             	"col
309879" NULLS LAST, col309880 DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_affine(st_pointfromgeohash('G'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY::GEOMETRY, 8.0::FLOAT8::FLOAT8::FLOAT8, (((7.0::FLOAT8::FLOAT8::FLOAT8 * 1.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 * 9.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 / 8.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8, (8.0::FLOAT8::FLOAT8::FLOAT8 + 3.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, (COALESCE(1.0::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8)::FLOAT8 * 6.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "col
309879",
        test_1        |             			NULL::TIME[] AS col309880
        test_1        |             	)
        test_1        |             		AS tab170183
        test_1        |             ORDER BY
        test_1        |             	"col
309879" NULLS LAST, col309880 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (841.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12090693?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12090693?buildTab=artifacts#/) on release-23.1 @ [cfadb904a838f5bc6a402b034501a0d7cee38cd8](https://github.com/cockroachdb/cockroach/commits/cfadb904a838f5bc6a402b034501a0d7cee38cd8):


```
        cockroach1_1  | status:              initialized new cluster
        cockroach1_1  | nodeID:              1
        postgres_1    | CREATE EXTENSION
        postgres_1    | Loading PostGIS extensions into postgres
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-10-07 05:17:37.101 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2023-10-07 05:17:37.103 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-10-07 05:17:37.105 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-10-07 05:17:37.107 UTC [51] LOG:  shutting down
        postgres_1    | 2023-10-07 05:17:37.197 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-10-07 05:17:37.222 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-10-07 05:17:37.222 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-10-07 05:17:37.222 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-10-07 05:17:37.225 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-10-07 05:17:37.230 UTC [68] LOG:  database system was shut down at 2023-10-07 05:17:37 UTC
        postgres_1    | 2023-10-07 05:17:37.235 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-10-07 05:17:39.360 UTC [78] ERROR:  foreign key constraint "t%qable1_col1_7_fkey" cannot be implemented
        postgres_1    | 2023-10-07 05:17:39.360 UTC [78] DETAIL:  Key columns "col1_7" and "col2_1" are of incompatible types: character varying and "char".
        postgres_1    | 2023-10-07 05:17:39.360 UTC [78] STATEMENT:  ALTER TABLE "t%qable1" ADD FOREIGN KEY (col1_7) REFERENCES "taBle2" (col2_1) ON DELETE CASCADE ON UPDATE RESTRICT
        test_1        | --- FAIL: TestCompare (606.03s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (1.22s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: ALTER TABLE "t%qable1" ADD FOREIGN KEY (col1_7) REFERENCES "taBle2" (col2_1) ON DELETE CASCADE ON UPDATE RESTRICT
        test_1        |         compare_test.go:173: postgres: exec: ERROR: foreign key constraint "t%qable1_col1_7_fkey" cannot be implemented (SQLSTATE 42804)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12137056?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12137056?buildTab=artifacts#/) on release-23.1 @ [30cd7fc605e1d05f4a37750806052625d7b9fae1](https://github.com/cockroachdb/cockroach/commits/30cd7fc605e1d05f4a37750806052625d7b9fae1):


```
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1323643892/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("POINT Z (0 2 0)"),
        test_1        |             + 	string("POINT(0 2 0)"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			CASE
        test_1        |             			WHEN false::BOOL THEN 'C':::STRING::STRING
        test_1        |             			ELSE st_asewkt(st_force3dz(COALESCE(st_point(0.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY, NULL::GEOMETRY)::GEOMETRY)::GEOMETRY::GEOMETRY)::STRING
        test_1        |             			END
        test_1        |             				AS "col%p109792"
        test_1        |             	)
        test_1        |             		AS " tab125848"
        test_1        |             ORDER BY
        test_1        |             	"col%p109792" NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			CASE
        test_1        |             			WHEN false::BOOL THEN 'C'::TEXT::TEXT
        test_1        |             			ELSE st_asewkt(st_force3dz(COALESCE(st_point(0.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY, NULL::GEOMETRY)::GEOMETRY)::GEOMETRY::GEOMETRY)::TEXT
        test_1        |             			END
        test_1        |             				AS "col%p109792"
        test_1        |             	)
        test_1        |             		AS " tab125848"
        test_1        |             ORDER BY
        test_1        |             	"col%p109792" NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (761.45s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12148779?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12148779?buildTab=artifacts#/) on release-23.1 @ [1d5f07d043035d14735c328a59384cd96cb47b09](https://github.com/cockroachdb/cockroach/commits/1d5f07d043035d14735c328a59384cd96cb47b09):


```
        postgres_1    | 2023-10-12 05:17:13.334 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-10-12 05:17:13.334 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-10-12 05:17:13.334 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-10-12 05:17:13.338 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-10-12 05:17:13.343 UTC [68] LOG:  database system was shut down at 2023-10-12 05:17:13 UTC
        postgres_1    | 2023-10-12 05:17:13.349 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-10-12 05:27:16.489 UTC [98] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-10-12 05:27:16.489 UTC [98] STATEMENT:  CREATE TABLE "t ab😍le
        postgres_1    | 	2" (col2_0 FLOAT4[] NOT NULL, col2_1 GEOGRAPHY NOT NULL, col2_2 TIMESTAMP NOT NULL, "c͡\\x9aol2_3" INT8 NOT NULL, "co l2_4" DATE, col2_5 TEXT NOT NULL, "col�2_6" TIMESTAMP NOT NULL, col2_7 BOOL NOT NULL, "col2_%p8" DATE NOT NULL, col2_9 TEXT NOT NULL, "col2 _10" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN 't15~$x)'::TEXT ELSE ''::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN e'U\x0e\x0f\x13=hz'::TEXT ELSE e'\x07'::TEXT END) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) STORED, "c
        postgres_1    | 	ol2_13" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_2 IS NULL THEN e'+BX76<9\x05K'::TEXT ELSE e'\'\''::TEXT END) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_2 IS NULL THEN e'\x1aE^F)k\x054A'::TEXT ELSE NULL END) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2_%p8" IS NULL THEN e'\x10%'::TEXT ELSE e'~&\x02iX'::TEXT END) STORED, "-col2_16" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2_%p8" IS NULL THEN e'``\t\x14\ff.f F'::TEXT ELSE 'exo'::TEXT END) STORED, col2_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_7 AS TEXT))) STORED, PRIMARY KEY (col2_11, "c
        postgres_1    | 	ol2_13", "col�2_6", col2_14, col2_15, "-col2_16", col2_0, col2_12, col2_9, col2_7, "col2_%p8", "col2 _10", col2_2, col2_5));
        postgres_1    | 	CREATE UNIQUE INDEX ON "t ab😍le
        postgres_1    | 	2" (col2_5, "col�2_6" ASC, col2_12 ASC) INCLUDE ("c͡\\x9aol2_3", "co l2_4");
        postgres_1    | 	CREATE UNIQUE INDEX ON "t ab😍le
        postgres_1    | 	2" (col2_12 DESC, col2_15 ASC, (CASE WHEN col2_0 IS NULL THEN e'{\x17>dq%'::TEXT ELSE e'\x19'::TEXT END), "-col2_16" ASC, col2_11 DESC, col2_7 ASC, "co l2_4" ASC, col2_17);
        postgres_1    | 	CREATE INDEX ON "t ab😍le
        postgres_1    | 	2" (col2_11, "-col2_16" ASC, col2_9, col2_0 DESC, col2_12 DESC, "col�2_6" ASC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.46s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.55s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "t ab😍le
        test_1        |             2" (col2_0 FLOAT4[] NOT NULL, col2_1 GEOGRAPHY NOT NULL, col2_2 TIMESTAMP NOT NULL, "c͡\\x9aol2_3" INT8 NOT NULL, "co l2_4" DATE, col2_5 TEXT NOT NULL, "col�2_6" TIMESTAMP NOT NULL, col2_7 BOOL NOT NULL, "col2_%p8" DATE NOT NULL, col2_9 TEXT NOT NULL, "col2 _10" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN 't15~$x)'::TEXT ELSE ''::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN e'U\x0e\x0f\x13=hz'::TEXT ELSE e'\x07'::TEXT END) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) STORED, "c
        test_1        |             ol2_13" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_2 IS NULL THEN e'+BX76<9\x05K'::TEXT ELSE e'\'\''::TEXT END) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_2 IS NULL THEN e'\x1aE^F)k\x054A'::TEXT ELSE NULL END) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2_%p8" IS NULL THEN e'\x10%'::TEXT ELSE e'~&\x02iX'::TEXT END) STORED, "-col2_16" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2_%p8" IS NULL THEN e'``\t\x14\ff.f F'::TEXT ELSE 'exo'::TEXT END) STORED, col2_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_7 AS TEXT))) STORED, PRIMARY KEY (col2_11, "c
        test_1        |             ol2_13", "col�2_6", col2_14, col2_15, "-col2_16", col2_0, col2_12, col2_9, col2_7, "col2_%p8", "col2 _10", col2_2, col2_5));
        test_1        |             CREATE UNIQUE INDEX ON "t ab😍le
        test_1        |             2" (col2_5, "col�2_6" ASC, col2_12 ASC) INCLUDE ("c͡\\x9aol2_3", "co l2_4");
        test_1        |             CREATE UNIQUE INDEX ON "t ab😍le
        test_1        |             2" (col2_12 DESC, col2_15 ASC, (CASE WHEN col2_0 IS NULL THEN e'{\x17>dq%'::TEXT ELSE e'\x19'::TEXT END), "-col2_16" ASC, col2_11 DESC, col2_7 ASC, "co l2_4" ASC, col2_17);
        test_1        |             CREATE INDEX ON "t ab😍le
        test_1        |             2" (col2_11, "-col2_16" ASC, col2_9, col2_0 DESC, col2_12 DESC, "col�2_6" ASC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12162081?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12162081?buildTab=artifacts#/) on release-23.1 @ [1d5f07d043035d14735c328a59384cd96cb47b09](https://github.com/cockroachdb/cockroach/commits/1d5f07d043035d14735c328a59384cd96cb47b09):


```
        postgres_1    | 							"t""a.%pb102390".col1_4 ASC
        postgres_1    | 					)
        postgres_1    | 		)
        postgres_1    | 			AS "ta&b102391"
        postgres_1    | 	ORDER BY
        postgres_1    | 		col190217 DESC NULLS FIRST, col190218 DESC NULLS FIRST
        test_1        | --- FAIL: TestCompare (714.61s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (110.70s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare963730227/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("0.0699268119435104G"),
        test_1        |             + 	string("0.06992681194351043G"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (tand(4.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 || 'G':::STRING::STRING::STRING)::STRING AS "Col190232")
        test_1        |             		AS tab102394
        test_1        |             ORDER BY
        test_1        |             	"Col190232" ASC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (tand(4.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 || 'G'::TEXT::TEXT::TEXT)::TEXT AS "Col190232")
        test_1        |             		AS tab102394
        test_1        |             ORDER BY
        test_1        |             	"Col190232" ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (738.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12185360?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12185360?buildTab=artifacts#/) on release-23.1 @ [366fe013b4b63bef98c953d03b302dc5e0b13ee7](https://github.com/cockroachdb/cockroach/commits/366fe013b4b63bef98c953d03b302dc5e0b13ee7):


```
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table4 AS tab436450
        postgres_1    | 			WHERE
        postgres_1    | 				NULL::BOOL
        postgres_1    | 		)
        postgres_1    | 			AS tab436451
        postgres_1    | 	ORDER BY
        postgres_1    | 		"c%7fol}805376" DESC, col805377 DESC NULLS LAST, " c ol805378" DESC NULLS LAST
        test_1        | --- FAIL: TestCompare (1167.89s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (562.96s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1898697998/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("0103000000010000000500000000000000008046C0000000000080464000000000008046C00000000000805640000000000000000000000000008056400000000000000000000000000080464000000000008046C00000000000804640"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('G':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS "co l805379") AS tab436452
        test_1        |             ORDER BY
        test_1        |             	"co l805379" ASC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('G'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS "co l805379") AS tab436452
        test_1        |             ORDER BY
        test_1        |             	"co l805379" ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1192.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12211550?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12211550?buildTab=artifacts#/) on release-23.1 @ [67bb386635d1dde8095e1c62f95aceb875616d49](https://github.com/cockroachdb/cockroach/commits/67bb386635d1dde8095e1c62f95aceb875616d49):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3393269208/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	nil,
        test_1        |             - 	string("00:00:09e"),
        test_1        |             + 	string("00:00:09E"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_4 AS col83876,
        test_1        |             			initcap(('00:00:09':::INTERVAL::INTERVAL::INTERVAL || 'E':::STRING::STRING::STRING)::STRING::STRING)::STRING
        test_1        |             				AS col83877
        test_1        |             	)
        test_1        |             		AS tab49094
        test_1        |             ORDER BY
        test_1        |             	col83876 DESC NULLS FIRST, col83877 DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_4 AS col83876,
        test_1        |             			initcap(('00:00:09'::INTERVAL::INTERVAL::INTERVAL || 'E'::TEXT::TEXT::TEXT)::TEXT::TEXT)::TEXT
        test_1        |             				AS col83877
        test_1        |             	)
        test_1        |             		AS tab49094
        test_1        |             ORDER BY
        test_1        |             	col83876 DESC NULLS FIRST, col83877 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (691.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12247693?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12247693?buildTab=artifacts#/) on release-23.1 @ [c04478348d7a7e11fb05b434264fe533bdad0881](https://github.com/cockroachdb/cockroach/commits/c04478348d7a7e11fb05b434264fe533bdad0881):


```
        postgres_1    | 				"tab|162163"."\gcol4_0",
        postgres_1    | 				"tab|162163"."co%l4/_6"
        postgres_1    | 			HAVING
        postgres_1    | 				every(true::BOOL::BOOL)::BOOL
        postgres_1    | 		)
        postgres_1    | 			AS tab162164
        postgres_1    | 	ORDER BY
        postgres_1    | 		col300182 ASC NULLS LAST, col300183 NULLS LAST
        test_1        | --- FAIL: TestCompare (791.10s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (186.82s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare185980318/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int32(79),
        test_1        |             + 	int32(78),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(atand(5.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col300276) AS "ta😱b""162223"
        test_1        |             ORDER BY
        test_1        |             	col300276 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(atand(5.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col300276) AS "ta😱b""162223"
        test_1        |             ORDER BY
        test_1        |             	col300276 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (815.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12265123?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12265123?buildTab=artifacts#/) on release-23.1 @ [ff444373b429926c6686eae90ad2fcc9551c4dff](https://github.com/cockroachdb/cockroach/commits/ff444373b429926c6686eae90ad2fcc9551c4dff):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | 2023-10-20 05:16:48.981 UTC [49] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2023-10-20 05:16:48.983 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-10-20 05:16:48.984 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-10-20 05:16:48.986 UTC [51] LOG:  shutting down
        postgres_1    | 2023-10-20 05:16:49.077 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-10-20 05:16:49.102 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-10-20 05:16:49.102 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-10-20 05:16:49.102 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-10-20 05:16:49.105 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-10-20 05:16:49.110 UTC [68] LOG:  database system was shut down at 2023-10-20 05:16:49 UTC
        postgres_1    | 2023-10-20 05:16:49.114 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-10-20 05:16:50.105 UTC [78] ERROR:  invalid input syntax for type numeric: "-Infinity" at character 666
        postgres_1    | 2023-10-20 05:16:50.105 UTC [78] STATEMENT:  CREATE TABLE "\vtable4" (col4_0 TEXT NOT NULL, col4_1 DECIMAL NOT NULL, col4_2 INET NULL, "col4(""_3" VARCHAR NOT NULL, _col4_4 TIMESTAMP NOT NULL, col4_5 REGTYPE NOT NULL, "col4_́'6" TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_0)) STORED, "Col4_""7" TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_0)) STORED, "col 4_8" DECIMAL NOT NULL GENERATED ALWAYS AS (col4_1 + (-20.3673765570856152)::DECIMAL) STORED, "�c̝ol4_9" TEXT NULL GENERATED ALWAYS AS (lower(CAST(col4_2 AS TEXT))) STORED, PRIMARY KEY (col4_5, "col4(""_3", _col4_4));
        postgres_1    | 	CREATE INDEX ON "\vtable4" ("col 4_8" ASC, "col4(""_3", col4_5 ASC, col4_2, _col4_4 DESC, "�c̝ol4_9", "col4_́'6", col4_0, (col4_1 + '-Infinity'::DECIMAL), "Col4_""7");
        postgres_1    | 	CREATE INDEX ON "\vtable4" ("�c̝ol4_9" DESC, "col4(""_3" DESC, col4_5 ASC, "Col4_""7" DESC, "col4_́'6", col4_0, _col4_4 ASC, col4_2 ASC) INCLUDE (col4_1);
        postgres_1    | 	CREATE UNIQUE INDEX ON "\vtable4" ("�c̝ol4_9" ASC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.02s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.94s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "\vtable4" (col4_0 TEXT NOT NULL, col4_1 DECIMAL NOT NULL, col4_2 INET NULL, "col4(""_3" VARCHAR NOT NULL, _col4_4 TIMESTAMP NOT NULL, col4_5 REGTYPE NOT NULL, "col4_́'6" TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_0)) STORED, "Col4_""7" TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_0)) STORED, "col 4_8" DECIMAL NOT NULL GENERATED ALWAYS AS (col4_1 + (-20.3673765570856152)::DECIMAL) STORED, "�c̝ol4_9" TEXT NULL GENERATED ALWAYS AS (lower(CAST(col4_2 AS TEXT))) STORED, PRIMARY KEY (col4_5, "col4(""_3", _col4_4));
        test_1        |             CREATE INDEX ON "\vtable4" ("col 4_8" ASC, "col4(""_3", col4_5 ASC, col4_2, _col4_4 DESC, "�c̝ol4_9", "col4_́'6", col4_0, (col4_1 + '-Infinity'::DECIMAL), "Col4_""7");
        test_1        |             CREATE INDEX ON "\vtable4" ("�c̝ol4_9" DESC, "col4(""_3" DESC, col4_5 ASC, "Col4_""7" DESC, "col4_́'6", col4_0, _col4_4 ASC, col4_2 ASC) INCLUDE (col4_1);
        test_1        |             CREATE UNIQUE INDEX ON "\vtable4" ("�c̝ol4_9" ASC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid input syntax for type numeric: "-Infinity" (SQLSTATE 22P02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (628.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12281299?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12281299?buildTab=artifacts#/) on release-23.1 @ [caaad29e5f92d11d926d2d9d8eb78b150e795e4d](https://github.com/cockroachdb/cockroach/commits/caaad29e5f92d11d926d2d9d8eb78b150e795e4d):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (570.90s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3555020300/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000001040000000000000224000000000000010400000000000000000000000000000104000000000000000000000000000001040000000000000224000000000000010400000000000002240"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(4.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "col 740313"
        test_1        |             	)
        test_1        |             		AS " tab429722"
        test_1        |             ORDER BY
        test_1        |             	"col 740313" ASC NULLS LAST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(4.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "col 740313"
        test_1        |             	)
        test_1        |             		AS " tab429722"
        test_1        |             ORDER BY
        test_1        |             	"col 740313" ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1200.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12292132?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12292132?buildTab=artifacts#/) on release-23.1 @ [caaad29e5f92d11d926d2d9d8eb78b150e795e4d](https://github.com/cockroachdb/cockroach/commits/caaad29e5f92d11d926d2d9d8eb78b150e795e4d):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (84.53s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare443998404/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("E0.714285714285714"),
        test_1        |             + 	string("E0.7142857142857143"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('E'::TEXT::TEXT::TEXT || (5.0::FLOAT8::FLOAT8::FLOAT8 / 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::TEXT
        test_1        |             				AS "cOl{133491"
        test_1        |             	)
        test_1        |             		AS tab68598
        test_1        |             ORDER BY
        test_1        |             	"cOl{133491" NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('E':::STRING::STRING::STRING || (5.0:::FLOAT8::FLOAT8::FLOAT8 / 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::STRING
        test_1        |             				AS "cOl{133491"
        test_1        |             	)
        test_1        |             		AS tab68598
        test_1        |             ORDER BY
        test_1        |             	"cOl{133491" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (713.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12320947?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12320947?buildTab=artifacts#/) on release-23.1 @ [942772ece66949001713bf1f41aaa97bba057cba](https://github.com/cockroachdb/cockroach/commits/942772ece66949001713bf1f41aaa97bba057cba):


```
        postgres_1    | 		"col 593273" NULLS FIRST,
        postgres_1    | 		"cOl593274" DESC NULLS LAST,
        postgres_1    | 		col593275 ASC,
        postgres_1    | 		col593276 DESC NULLS FIRST
        postgres_1    | 2023-10-24 05:24:11.066 UTC [78] ERROR:  collation "de_de" for encoding "UTF8" does not exist at character 34
        postgres_1    | 2023-10-24 05:24:11.066 UTC [78] STATEMENT:  SELECT * FROM (SELECT NULL::TEXT COLLATE de_DE AS col593285) AS "ta b290696" ORDER BY col593285 DESC
        test_1        | --- FAIL: TestCompare (1009.14s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (403.47s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1270075951/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("1C"),
        test_1        |             + 	string("1.0000000000000000C"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (exp(0:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'C':::STRING::STRING::STRING)::STRING AS col593375)
        test_1        |             		AS "%vtaB290728"
        test_1        |             ORDER BY
        test_1        |             	col593375 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (exp(0::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'C'::TEXT::TEXT::TEXT)::TEXT AS col593375)
        test_1        |             		AS "%vtaB290728"
        test_1        |             ORDER BY
        test_1        |             	col593375 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1033.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12340154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12340154?buildTab=artifacts#/) on release-23.1 @ [7d70ac8dccaf3035ecf8c6f351b43d54fe6a4e26](https://github.com/cockroachdb/cockroach/commits/7d70ac8dccaf3035ecf8c6f351b43d54fe6a4e26):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (346.30s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3687848719/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("01030000000100000005000000000000000000004000000000000000400000000000000040000000000000F03F0000000000001C40000000000000F03F0000000000001C40000000000000004000000000000000400000000000000040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(2.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col356665
        test_1        |             	)
        test_1        |             		AS tab209852
        test_1        |             ORDER BY
        test_1        |             	col356665 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(2.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col356665
        test_1        |             	)
        test_1        |             		AS tab209852
        test_1        |             ORDER BY
        test_1        |             	col356665 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (975.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12359422?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12359422?buildTab=artifacts#/) on release-23.1 @ [310f003cc3a1970ca5390bac408ed72e6e7ad996](https://github.com/cockroachdb/cockroach/commits/310f003cc3a1970ca5390bac408ed72e6e7ad996):


```
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table1 AS tab368462
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab368462.tableoid NULLS FIRST, tab368462.col1_0 DESC
        postgres_1    | 		)
        postgres_1    | 			AS tab368463
        postgres_1    | 	ORDER BY
        postgres_1    | 		col321715 NULLS FIRST, "co%pl321716" NULLS FIRST
        test_1        | --- FAIL: TestCompare (1024.06s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (419.38s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare351092592/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int32(202),
        test_1        |             + 	int32(201),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(sinh(6.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS 😭col321721) AS "t
ab368468"
        test_1        |             ORDER BY
        test_1        |             	😭col321721 ASC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(sinh(6.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS 😭col321721) AS "t
ab368468"
        test_1        |             ORDER BY
        test_1        |             	😭col321721 ASC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1048.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12375800?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12375800?buildTab=artifacts#/) on release-23.1 @ [7eb95880d348e5e65b1cb133b9d7d505a035b078](https://github.com/cockroachdb/cockroach/commits/7eb95880d348e5e65b1cb133b9d7d505a035b078):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (167.43s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3521436613/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("G2.5000000000000000000"),
        test_1        |             + 	string("G2.5000000000000000"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('G':::STRING::STRING::STRING || (5:::INT8::INT8::INT8 / 2:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::STRING
        test_1        |             				AS col214078
        test_1        |             	)
        test_1        |             		AS "taB136710"
        test_1        |             ORDER BY
        test_1        |             	col214078 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('G'::TEXT::TEXT::TEXT || (5::INT8::INT8::INT8 / 2::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::TEXT
        test_1        |             				AS col214078
        test_1        |             	)
        test_1        |             		AS "taB136710"
        test_1        |             ORDER BY
        test_1        |             	col214078 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (796.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12391453?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12391453?buildTab=artifacts#/) on release-23.1 @ [bd8bf50082a06dc4c1e8defbcd4a7357f7e5d1f7](https://github.com/cockroachdb/cockroach/commits/bd8bf50082a06dc4c1e8defbcd4a7357f7e5d1f7):


```
        postgres_1    | 2023-10-28 05:16:35.344 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-10-28 05:16:35.368 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-10-28 05:16:35.368 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-10-28 05:16:35.368 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-10-28 05:16:35.371 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-10-28 05:16:35.376 UTC [68] LOG:  database system was shut down at 2023-10-28 05:16:35 UTC
        postgres_1    | 2023-10-28 05:16:35.380 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-10-28 05:16:36.353 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-10-28 05:16:36.353 UTC [78] STATEMENT:  CREATE TABLE table1 ("col1 _0" INET NOT NULL, "col1!_1" TEXT NOT NULL, col1_2 TEXT NOT NULL, "col1'_3" INET NOT NULL, "co�l_1%v_4" REGCLASS NOT NULL, col1_5 REGCLASS NOT NULL, "col1%56_6" INT2 NOT NULL, col1__7 REGCLASS NOT NULL, col1_8 INTERVAL NOT NULL, col1_9 BOOL NOT NULL, col1_10 UUID NOT NULL, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (lower("col1!_1")) STORED, "col1 _12" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST("col1 _0" AS TEXT))) STORED, PRIMARY KEY (col1_2, "col1 _12", col1_11, col1_8, "co�l_1%v_4", "col1 _0", col1__7, col1_9, col1_5, "col1'_3", "col1!_1", "col1%56_6", col1_10));
        postgres_1    | 	CREATE INDEX ON table1 ("col1%56_6" DESC, col1__7, col1_5 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_11, "col1'_3" DESC, col1_2, col1_10 ASC, (CASE WHEN "co�l_1%v_4" IS NULL THEN e'8\x15b\f'::TEXT ELSE 'e'::TEXT END) DESC, "col1!_1", "col1 _12");
        postgres_1    | 	CREATE INDEX ON table1 (col1_8 DESC, "col1 _12" DESC, col1_5 ASC, col1_2, col1_11);
        postgres_1    | 	CREATE INDEX ON table1 (col1_10, "col1 _0" DESC, (CASE WHEN "co�l_1%v_4" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) DESC, col1_2 ASC, "co�l_1%v_4" ASC, lower(CAST("col1 _0" AS TEXT)) DESC, col1_8, "col1 _12", col1__7 DESC, col1_5 ASC, "col1%56_6", col1_9);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_5 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_10, "col1 _12" DESC, "col1%56_6" ASC, col1_8);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.40s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.87s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table1 ("col1 _0" INET NOT NULL, "col1!_1" TEXT NOT NULL, col1_2 TEXT NOT NULL, "col1'_3" INET NOT NULL, "co�l_1%v_4" REGCLASS NOT NULL, col1_5 REGCLASS NOT NULL, "col1%56_6" INT2 NOT NULL, col1__7 REGCLASS NOT NULL, col1_8 INTERVAL NOT NULL, col1_9 BOOL NOT NULL, col1_10 UUID NOT NULL, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (lower("col1!_1")) STORED, "col1 _12" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST("col1 _0" AS TEXT))) STORED, PRIMARY KEY (col1_2, "col1 _12", col1_11, col1_8, "co�l_1%v_4", "col1 _0", col1__7, col1_9, col1_5, "col1'_3", "col1!_1", "col1%56_6", col1_10));
        test_1        |             CREATE INDEX ON table1 ("col1%56_6" DESC, col1__7, col1_5 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_11, "col1'_3" DESC, col1_2, col1_10 ASC, (CASE WHEN "co�l_1%v_4" IS NULL THEN e'8\x15b\f'::TEXT ELSE 'e'::TEXT END) DESC, "col1!_1", "col1 _12");
        test_1        |             CREATE INDEX ON table1 (col1_8 DESC, "col1 _12" DESC, col1_5 ASC, col1_2, col1_11);
        test_1        |             CREATE INDEX ON table1 (col1_10, "col1 _0" DESC, (CASE WHEN "co�l_1%v_4" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) DESC, col1_2 ASC, "co�l_1%v_4" ASC, lower(CAST("col1 _0" AS TEXT)) DESC, col1_8, "col1 _12", col1__7 DESC, col1_5 ASC, "col1%56_6", col1_9);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_5 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_10, "col1 _12" DESC, "col1%56_6" ASC, col1_8);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12412096?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12412096?buildTab=artifacts#/) on release-23.1 @ [0a04f59b9d15987525620fb047fc0c3d65854738](https://github.com/cockroachdb/cockroach/commits/0a04f59b9d15987525620fb047fc0c3d65854738):


```
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	[]uint8{
        test_1        |               		0x81,
        test_1        |               		0x08,
        test_1        |             - 		0x09,
        test_1        |             + 		0x69,
        test_1        |               		0x00,
        test_1        |               		0x80,
        test_1        |               		... // 4 identical elements
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepoint(0.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 4:::INT8::INT4::INT4, 2:::INT8::INT4::INT4, 3:::INT8::INT4::INT4)::BYTES
        test_1        |             				AS "col'159347"
        test_1        |             	)
        test_1        |             		AS "t̯a,b110256"
        test_1        |             ORDER BY
        test_1        |             	"col'159347";
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepoint(0.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 4::INT8::INT4::INT4, 2::INT8::INT4::INT4, 3::INT8::INT4::INT4)::BYTEA
        test_1        |             				AS "col'159347"
        test_1        |             	)
        test_1        |             		AS "t̯a,b110256"
        test_1        |             ORDER BY
        test_1        |             	"col'159347";
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (783.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12498436?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12498436?buildTab=artifacts#/) on release-23.1 @ [239099ef36e3f729d52ff1b5690d1452e07299c6](https://github.com/cockroachdb/cockroach/commits/239099ef36e3f729d52ff1b5690d1452e07299c6):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (556.96s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1890274590/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("::7/0I"),
        test_1        |             + 	string("::7/0i"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			initcap(('::7/0':::INET::INET::INET || 'I':::STRING::STRING::STRING)::STRING::STRING)::STRING
        test_1        |             				AS "col\g733207"
        test_1        |             	)
        test_1        |             		AS tab413432
        test_1        |             ORDER BY
        test_1        |             	"col\g733207" NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			initcap(('::7/0'::INET::INET::INET || 'I'::TEXT::TEXT::TEXT)::TEXT::TEXT)::TEXT
        test_1        |             				AS "col\g733207"
        test_1        |             	)
        test_1        |             		AS tab413432
        test_1        |             ORDER BY
        test_1        |             	"col\g733207" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1187.24s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #111157 compose: TestComposeCompare failed [C-test-failure FROZEN.branch-release-23.1.11-rc O-robot O-rsg T-sql-queries]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12584567?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12584567?buildTab=artifacts#/) on release-23.1 @ [2f6bd0a1dec77b7a4c947ec7dc1bf250b7038d05](https://github.com/cockroachdb/cockroach/commits/2f6bd0a1dec77b7a4c947ec7dc1bf250b7038d05):


```
        postgres_1    | 	ORDER BY
        postgres_1    | 		col646016,
        postgres_1    | 		col646017 ASC NULLS LAST,
        postgres_1    | 		c😗ol646018 ASC NULLS LAST,
        postgres_1    | 		"col%q646019" ASC NULLS LAST,
        postgres_1    | 		"c\\U001095A9ol646020"
        test_1        | --- FAIL: TestCompare (1036.30s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (431.26s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare4001203848/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("010300000001000000050000000000000000E060C000000000008046400000000000E060C0000000000080564000000000008056C0000000000080564000000000008056C000000000008046400000000000E060C00000000000804640"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('C':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS "{col646026")
        test_1        |             		AS "t-ab368720"
        test_1        |             ORDER BY
        test_1        |             	"{col646026" ASC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('C'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS "{col646026")
        test_1        |             		AS "t-ab368720"
        test_1        |             ORDER BY
        test_1        |             	"{col646026" ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1061.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12667652?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12667652?buildTab=artifacts#/) on release-23.1 @ [a612b60e1ee2ac0b38fc7735b01715be0cbb790e](https://github.com/cockroachdb/cockroach/commits/a612b60e1ee2ac0b38fc7735b01715be0cbb790e):


```
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare4038060239/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("00:02:00B1.2857142857142858"),
        test_1        |             + 	string("00:02:00B1.28571428571429"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(('00:02:00':::TIME::TIME::TIME || 'B':::STRING::STRING::STRING)::STRING::STRING || (9.0:::FLOAT8::FLOAT8::FLOAT8 / 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::STRING
        test_1        |             				AS "col
        test_1        |             104802"
        test_1        |             	)
        test_1        |             		AS " tab58524"
        test_1        |             ORDER BY
        test_1        |             	"col
        test_1        |             104802" NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(('00:02:00'::TIME::TIME::TIME || 'B'::TEXT::TEXT::TEXT)::TEXT::TEXT || (9.0::FLOAT8::FLOAT8::FLOAT8 / 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::TEXT
        test_1        |             				AS "col
        test_1        |             104802"
        test_1        |             	)
        test_1        |             		AS " tab58524"
        test_1        |             ORDER BY
        test_1        |             	"col
        test_1        |             104802" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (712.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12686595?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12686595?buildTab=artifacts#/) on release-23.1 @ [1700413c09ff1cc391e1fb0560f8f6d8e5258d19](https://github.com/cockroachdb/cockroach/commits/1700413c09ff1cc391e1fb0560f8f6d8e5258d19):


```
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1031714871/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(2),
        test_1        |             + 	int16(3),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			CASE
        test_1        |             			WHEN false::BOOL THEN NULL::INT2
        test_1        |             			ELSE int2(sqrt(7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2
        test_1        |             			END
        test_1        |             				AS "%vcol'598333"
        test_1        |             	)
        test_1        |             		AS tab397979
        test_1        |             ORDER BY
        test_1        |             	"%vcol'598333" DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			CASE
        test_1        |             			WHEN false::BOOL THEN NULL::INT2
        test_1        |             			ELSE int2(sqrt(7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2
        test_1        |             			END
        test_1        |             				AS "%vcol'598333"
        test_1        |             	)
        test_1        |             		AS tab397979
        test_1        |             ORDER BY
        test_1        |             	"%vcol'598333" DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1170.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12704369?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12704369?buildTab=artifacts#/) on release-23.1 @ [0c43ef7d21f6fa375b8a7f6088884f2fdf10952d](https://github.com/cockroachdb/cockroach/commits/0c43ef7d21f6fa375b8a7f6088884f2fdf10952d):


```
        postgres_1    | 			AS tab231779
        postgres_1    | 	ORDER BY
        postgres_1    | 		"col(361408" DESC NULLS FIRST,
        postgres_1    | 		col361409 DESC NULLS FIRST,
        postgres_1    | 		col361410 ASC,
        postgres_1    | 		col361411 DESC NULLS LAST,
        postgres_1    | 		"co\\u2E40l361412" ASC,
        postgres_1    | 		CAST(😡col361413 AS TEXT) ASC NULLS FIRST
        test_1        | --- FAIL: TestCompare (878.45s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (274.55s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3532596115/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int32(2),
        test_1        |             + 	int32(3),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(asinh(9.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS "\\xf3co̧l361594") AS tab231876
        test_1        |             ORDER BY
        test_1        |             	"\\xf3co̧l361594" ASC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(asinh(9.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS "\\xf3co̧l361594") AS tab231876
        test_1        |             ORDER BY
        test_1        |             	"\\xf3co̧l361594" ASC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (902.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109969 compose: TestComposeCompare failed [FROZEN.branch-release-23.1.11-rc O-robot T-sql-foundations]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #108652 compose: TestComposeCompare failed [O-robot T-sql-foundations TODO-deprecate.branch-release-23.1.9-rc]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12722218?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12722218?buildTab=artifacts#/) on release-23.1 @ [85392ec3a02bf344ab55b8e8154b27f5ab3f9a45](https://github.com/cockroachdb/cockroach/commits/85392ec3a02bf344ab55b8e8154b27f5ab3f9a45):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (8.09s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3679753943/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("A2097152"),
        test_1        |             + 	string("A2097152.0000000000000000"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('A'::TEXT::TEXT::TEXT || (8::INT8::INT8::INT8 ^ 7::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::TEXT
        test_1        |             				AS "col\\u103B6278"
        test_1        |             	)
        test_1        |             		AS tab3341
        test_1        |             ORDER BY
        test_1        |             	"col\\u103B6278" NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('A':::STRING::STRING::STRING || (8:::INT8::INT8::INT8 ^ 7:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::STRING
        test_1        |             				AS "col\\u103B6278"
        test_1        |             	)
        test_1        |             		AS tab3341
        test_1        |             ORDER BY
        test_1        |             	"col\\u103B6278" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (636.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12737355?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12737355?buildTab=artifacts#/) on release-23.1 @ [3e8713f9ffd60e20cc58d2df27da17f60ffad24a](https://github.com/cockroachdb/cockroach/commits/3e8713f9ffd60e20cc58d2df27da17f60ffad24a):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (374.05s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare230154412/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	nil,
        test_1        |             - 	string("D3.7621956910836314"),
        test_1        |             + 	string("D3.76219569108363"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_2 AS col560577,
        test_1        |             			('D':::STRING::STRING::STRING || cosh(2.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::STRING AS "'col560578"
        test_1        |             	)
        test_1        |             		AS tab311842
        test_1        |             ORDER BY
        test_1        |             	col560577 DESC, "'col560578";
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_2 AS col560577,
        test_1        |             			('D'::TEXT::TEXT::TEXT || cosh(2.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::TEXT AS "'col560578"
        test_1        |             	)
        test_1        |             		AS tab311842
        test_1        |             ORDER BY
        test_1        |             	col560577 DESC, "'col560578";
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1002.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12772158?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12772158?buildTab=artifacts#/) on release-23.1 @ [e268b992676d69661540d196be7248a9369c7396](https://github.com/cockroachdb/cockroach/commits/e268b992676d69661540d196be7248a9369c7396):


```
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-11-21 06:17:18.178 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2023-11-21 06:17:18.181 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-11-21 06:17:18.182 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-11-21 06:17:18.184 UTC [51] LOG:  shutting down
        test_1        | I231121 06:17:18.233341 38 testutils/soon.go:41  [-] 1  SucceedsSoon: failed to connect to ``host=postgres user=postgres database=postgres``: dial error (dial tcp 172.18.0.3:5432: connect: connection refused)
        postgres_1    | 2023-11-21 06:17:18.292 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-11-21 06:17:18.401 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-11-21 06:17:18.401 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-11-21 06:17:18.401 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-11-21 06:17:18.405 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-11-21 06:17:18.410 UTC [68] LOG:  database system was shut down at 2023-11-21 06:17:18 UTC
        postgres_1    | 2023-11-21 06:17:18.415 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-11-21 06:17:20.325 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-11-21 06:17:20.325 UTC [78] STATEMENT:  CREATE TABLE table1 (col1_0 INTERVAL NOT NULL, "col1_'1" INT2 NOT NULL, "col1_*2" INTERVAL NOT NULL, col1_3 REGTYPE NOT NULL, "'col1_4" DATE NOT NULL, "col1\\u5324_5" INTERVAL NOT NULL, col1_6 FLOAT8 NOT NULL, "col1	_7" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1\\u5324_5" IS NULL THEN e'\U00002603'::TEXT ELSE e'\x0f,h30\x1cD(.'::TEXT END) STORED, col1_8 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN ''::TEXT ELSE e'\x04\x1dHh\x1b\x10\x18\x18\x1e'::TEXT END) STORED, col1_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN ' W '::TEXT ELSE e'-{>\x05]"P'::TEXT END) STORED, 😥col1_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_3 IS NULL THEN e'5Y\x11%bS.G='::TEXT ELSE 'G^'::TEXT END) STORED, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "'col1_4" IS NULL THEN e'\''::TEXT ELSE e'g_\x16>\x142'::TEXT END) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_3, col1_11 ASC, "col1_'1" DESC, "col1_*2" ASC) INCLUDE ("'col1_4");
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_6 DESC) INCLUDE (col1_3, "'col1_4", "col1\\u5324_5");
        postgres_1    | 	CREATE INDEX ON table1 (col1_3 DESC, "col1	_7" DESC, col1_0 ASC, (CASE WHEN col1_0 IS NULL THEN e'\x00'::TEXT ELSE e'7>\x1d\x05Fjf7\r'::TEXT END) DESC, 😥col1_10, col1_9, "col1\\u5324_5" DESC, (CASE WHEN "col1_*2" IS NULL THEN e'K\x02ybS;'::TEXT ELSE e'\''::TEXT END), col1_8 DESC) INCLUDE ("col1_*2", "'col1_4");
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.35s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.08s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table1 (col1_0 INTERVAL NOT NULL, "col1_'1" INT2 NOT NULL, "col1_*2" INTERVAL NOT NULL, col1_3 REGTYPE NOT NULL, "'col1_4" DATE NOT NULL, "col1\\u5324_5" INTERVAL NOT NULL, col1_6 FLOAT8 NOT NULL, "col1	_7" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1\\u5324_5" IS NULL THEN e'\U00002603'::TEXT ELSE e'\x0f,h30\x1cD(.'::TEXT END) STORED, col1_8 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN ''::TEXT ELSE e'\x04\x1dHh\x1b\x10\x18\x18\x1e'::TEXT END) STORED, col1_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN ' W '::TEXT ELSE e'-{>\x05]"P'::TEXT END) STORED, 😥col1_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_3 IS NULL THEN e'5Y\x11%bS.G='::TEXT ELSE 'G^'::TEXT END) STORED, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "'col1_4" IS NULL THEN e'\''::TEXT ELSE e'g_\x16>\x142'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3, col1_11 ASC, "col1_'1" DESC, "col1_*2" ASC) INCLUDE ("'col1_4");
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_6 DESC) INCLUDE (col1_3, "'col1_4", "col1\\u5324_5");
        test_1        |             CREATE INDEX ON table1 (col1_3 DESC, "col1	_7" DESC, col1_0 ASC, (CASE WHEN col1_0 IS NULL THEN e'\x00'::TEXT ELSE e'7>\x1d\x05Fjf7\r'::TEXT END) DESC, 😥col1_10, col1_9, "col1\\u5324_5" DESC, (CASE WHEN "col1_*2" IS NULL THEN e'K\x02ybS;'::TEXT ELSE e'\''::TEXT END), col1_8 DESC) INCLUDE ("col1_*2", "'col1_4");
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12790423?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12790423?buildTab=artifacts#/) on release-23.1 @ [a591a687999f6b73f5489628b49ae768ce6a425e](https://github.com/cockroachdb/cockroach/commits/a591a687999f6b73f5489628b49ae768ce6a425e):


```
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(power(2::DECIMAL::DECIMAL::DECIMAL, 7::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS col448601,
        test_1        |             			NULL::OIDVECTOR AS col448602
        test_1        |             	)
        test_1        |             		AS "tab'316981"
        test_1        |             ORDER BY
        test_1        |             	col448601 ASC NULLS FIRST, col448602 ASC;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(power(2:::DECIMAL::DECIMAL::DECIMAL, 7:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS col448601,
        test_1        |             			NULL::OIDVECTOR AS col448602
        test_1        |             	)
        test_1        |             		AS "tab'316981"
        test_1        |             ORDER BY
        test_1        |             	col448601 ASC NULLS FIRST, col448602 ASC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (942.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12832233?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12832233?buildTab=artifacts#/) on release-23.1 @ [0f97132ad066321e6f65a83fe2098788162445ae](https://github.com/cockroachdb/cockroach/commits/0f97132ad066321e6f65a83fe2098788162445ae):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare29346291/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	strings.Join({
        test_1        |               		"BOX(-90 0,-89.99995708465576 0.000042915344238",
        test_1        |             - 		"28125",
        test_1        |               		")",
        test_1        |               	}, ""),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash(('D':::STRING::STRING::STRING || '00000000-0000-0000-0000-000000000005':::UUID::UUID::UUID)::STRING::STRING, 9:::INT8::INT4::INT4)::BOX2D
        test_1        |             				AS "co l697537"
        test_1        |             	)
        test_1        |             		AS tab408632
        test_1        |             ORDER BY
        test_1        |             	CAST("co l697537" AS STRING) DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash(('D'::TEXT::TEXT::TEXT || '00000000-0000-0000-0000-000000000005'::UUID::UUID::UUID)::TEXT::TEXT, 9::INT8::INT4::INT4)::BOX2D
        test_1        |             				AS "co l697537"
        test_1        |             	)
        test_1        |             		AS tab408632
        test_1        |             ORDER BY
        test_1        |             	CAST("co l697537" AS TEXT) DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1213.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12867547?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12867547?buildTab=artifacts#/) on release-23.1 @ [3f356b85b529262b2bdda605c4b8e56598e29fae](https://github.com/cockroachdb/cockroach/commits/3f356b85b529262b2bdda605c4b8e56598e29fae):


```
        postgres_1    | 		col148030 NULLS LAST,
        postgres_1    | 		col148031 ASC NULLS LAST,
        postgres_1    | 		col148032 NULLS FIRST,
        postgres_1    | 		"col'148033" NULLS LAST,
        postgres_1    | 		col148034 ASC NULLS FIRST,
        postgres_1    | 		col148035 DESC NULLS FIRST,
        postgres_1    | 		col148036 DESC NULLS FIRST,
        postgres_1    | 		"c ol148037" DESC
        test_1        | --- FAIL: TestCompare (703.27s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (97.80s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare337989723/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(0),
        test_1        |             + 	int16(1),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(cosd(8.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "Col148054") AS "t\\u0E12🙃ab74682"
        test_1        |             ORDER BY
        test_1        |             	"Col148054" DESC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(cosd(8.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "Col148054") AS "t\\u0E12🙃ab74682"
        test_1        |             ORDER BY
        test_1        |             	"Col148054" DESC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (727.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12963614?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/12963614?buildTab=artifacts#/) on release-23.1 @ [1cc63e4605a447c4920c35c3f0197bd9bd9558df](https://github.com/cockroachdb/cockroach/commits/1cc63e4605a447c4920c35c3f0197bd9bd9558df):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2437642602/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("2097152E"),
        test_1        |             + 	string("2097152.0000000000000000E"),
        test_1        |               	int32(8),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(pow(8:::DECIMAL::DECIMAL::DECIMAL, 7:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'E':::STRING::STRING::STRING)::STRING
        test_1        |             				AS "cOl481016",
        test_1        |             			int4(8.0:::FLOAT8::FLOAT8::FLOAT8)::INT4 AS col481017
        test_1        |             	)
        test_1        |             		AS tab259576
        test_1        |             ORDER BY
        test_1        |             	"cOl481016" NULLS LAST, col481017 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(pow(8::DECIMAL::DECIMAL::DECIMAL, 7::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'E'::TEXT::TEXT::TEXT)::TEXT
        test_1        |             				AS "cOl481016",
        test_1        |             			int4(8.0::FLOAT8::FLOAT8::FLOAT8)::INT4 AS col481017
        test_1        |             	)
        test_1        |             		AS tab259576
        test_1        |             ORDER BY
        test_1        |             	"cOl481016" NULLS LAST, col481017 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1008.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13016962?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13016962?buildTab=artifacts#/) on release-23.1 @ [5643a5e33f67fb93f9fccaa3cbe8e8deddaeafe7](https://github.com/cockroachdb/cockroach/commits/5643a5e33f67fb93f9fccaa3cbe8e8deddaeafe7):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (36.16s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1126638583/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000000840000000000000F03F00000000000008400000000000000840000000000000000000000000000008400000000000000000000000000000F03F0000000000000840000000000000F03F"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0:::FLOAT8::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col53165
        test_1        |             	)
        test_1        |             		AS "tab%p31956"
        test_1        |             ORDER BY
        test_1        |             	col53165 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0::FLOAT8::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col53165
        test_1        |             	)
        test_1        |             		AS "tab%p31956"
        test_1        |             ORDER BY
        test_1        |             	col53165 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (664.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113814 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2.17-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13110963?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13110963?buildTab=artifacts#/) on release-23.1 @ [2dfb2cbae66bac7cd04782dccc9267bc2dc063b1](https://github.com/cockroachdb/cockroach/commits/2dfb2cbae66bac7cd04782dccc9267bc2dc063b1):


```
        postgres_1    | 			AS t̪ab409311
        postgres_1    | 	ORDER BY
        postgres_1    | 		co_l789427 ASC,
        postgres_1    | 		col789428 DESC NULLS LAST,
        postgres_1    | 		"col'789429" NULLS FIRST,
        postgres_1    | 		",col789430" DESC,
        postgres_1    | 		"c%po""l789431" DESC NULLS LAST,
        postgres_1    | 		"cOl789432" DESC NULLS LAST
        test_1        | --- FAIL: TestCompare (1196.54s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (591.57s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3703742167/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int32(3),
        test_1        |             + 	int32(4),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(cosh(2.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col789511) AS tab409375
        test_1        |             ORDER BY
        test_1        |             	col789511 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int4(cosh(2.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT4 AS col789511) AS tab409375
        test_1        |             ORDER BY
        test_1        |             	col789511 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1221.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116303 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1-merge-v23.1.13]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13152794?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13152794?buildTab=artifacts#/) on release-23.1 @ [db14a34ec09e0c9699357f327bce540c4a4f4b31](https://github.com/cockroachdb/cockroach/commits/db14a34ec09e0c9699357f327bce540c4a4f4b31):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-12-15 06:18:00.678 UTC [48] LOG:  received fast shutdown request
        postgres_1    | .2023-12-15 06:18:00.680 UTC [48] LOG:  aborting any active transactions
        postgres_1    | 2023-12-15 06:18:00.681 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
        postgres_1    | 2023-12-15 06:18:00.684 UTC [50] LOG:  shutting down
        postgres_1    | 2023-12-15 06:18:00.786 UTC [48] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-12-15 06:18:00.900 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-12-15 06:18:00.900 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-12-15 06:18:00.900 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-12-15 06:18:00.904 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-12-15 06:18:00.909 UTC [67] LOG:  database system was shut down at 2023-12-15 06:18:00 UTC
        postgres_1    | 2023-12-15 06:18:00.915 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-12-15 06:18:02.695 UTC [77] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-12-15 06:18:02.695 UTC [77] STATEMENT:  CREATE TABLE table1 (c͕ol1_0 TIMESTAMP NOT NULL, col1_1 BYTEA NOT NULL, col1_2 REGTYPE NOT NULL, col1_3 INT8 NOT NULL, "co*l1_ 4" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN c͕ol1_0 IS NULL THEN ''::TEXT ELSE ''::TEXT END) STORED, col1_5 INT8 GENERATED ALWAYS AS (col1_3 + (-4566245941947401852)::INT8) STORED, "
        postgres_1    | 	cO_l1_6" TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_1 IS NULL THEN '4/BrAYp'::TEXT ELSE e'\x00'::TEXT END) STORED, PRIMARY KEY (col1_3, col1_1, c͕ol1_0, col1_2));
        postgres_1    | 	CREATE INDEX ON table1 (col1_3 DESC, col1_2 ASC, "co*l1_ 4" DESC, "
        postgres_1    | 	cO_l1_6" DESC, col1_5 DESC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.16s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.01s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table1 (c͕ol1_0 TIMESTAMP NOT NULL, col1_1 BYTEA NOT NULL, col1_2 REGTYPE NOT NULL, col1_3 INT8 NOT NULL, "co*l1_ 4" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN c͕ol1_0 IS NULL THEN ''::TEXT ELSE ''::TEXT END) STORED, col1_5 INT8 GENERATED ALWAYS AS (col1_3 + (-4566245941947401852)::INT8) STORED, "
        test_1        |             cO_l1_6" TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_1 IS NULL THEN '4/BrAYp'::TEXT ELSE e'\x00'::TEXT END) STORED, PRIMARY KEY (col1_3, col1_1, c͕ol1_0, col1_2));
        test_1        |             CREATE INDEX ON table1 (col1_3 DESC, col1_2 ASC, "co*l1_ 4" DESC, "
        test_1        |             cO_l1_6" DESC, col1_5 DESC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116303 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1-merge-v23.1.13]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13172141?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13172141?buildTab=artifacts#/) on release-23.1 @ [d0e33ab98ad01af19d03b34da328825b48a1da7a](https://github.com/cockroachdb/cockroach/commits/d0e33ab98ad01af19d03b34da328825b48a1da7a):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (319.15s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3495525673/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("01010000000000000000205CC00000000000E05040"),
        test_1        |               	s"300000000",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_pointfromgeohash('C':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS "%cOl295172",
        test_1        |             			'00:05:00':::TIME::TIME AS col295173
        test_1        |             	)
        test_1        |             		AS " t'ab175767"
        test_1        |             ORDER BY
        test_1        |             	"%cOl295172" DESC NULLS LAST, col295173 ASC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_pointfromgeohash('C'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS "%cOl295172",
        test_1        |             			'00:05:00'::TIME::TIME AS col295173
        test_1        |             	)
        test_1        |             		AS " t'ab175767"
        test_1        |             ORDER BY
        test_1        |             	"%cOl295172" DESC NULLS LAST, col295173 ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (949.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13182683?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13182683?buildTab=artifacts#/) on release-23.1 @ [d0e33ab98ad01af19d03b34da328825b48a1da7a](https://github.com/cockroachdb/cockroach/commits/d0e33ab98ad01af19d03b34da328825b48a1da7a):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2023-12-17 06:17:10.199 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2023-12-17 06:17:10.200 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-12-17 06:17:10.202 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-12-17 06:17:10.204 UTC [51] LOG:  shutting down
        postgres_1    | 2023-12-17 06:17:10.302 UTC [49] LOG:  database system is shut down
        test_1        | I231217 06:17:10.316409 7 testutils/soon.go:41  [-] 1  SucceedsSoon: failed to connect to ``host=postgres user=postgres database=postgres``: dial error (dial tcp 172.18.0.3:5432: connect: connection refused)
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-12-17 06:17:10.421 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-12-17 06:17:10.421 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-12-17 06:17:10.421 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-12-17 06:17:10.425 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-12-17 06:17:10.430 UTC [68] LOG:  database system was shut down at 2023-12-17 06:17:10 UTC
        postgres_1    | 2023-12-17 06:17:10.436 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-12-17 06:17:12.452 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-12-17 06:17:12.452 UTC [78] STATEMENT:  CREATE TABLE "t""ab'le5" (col5_0 TIMESTAMP NOT NULL, "col5 _1" INT8 NOT NULL, col5_2 BYTEA NOT NULL, col5_3 JSONB NOT NULL, col5_4 INT2 NOT NULL, c͊ol5_5 TIME, "co"" %a1L5_\n6" JSONB NOT NULL, "%qcol5_7" TIME NOT NULL, col5_8 BOOL, "col5_\\x2c9" BYTEA NOT NULL, col5_10 JSONB NOT NULL, " col5_11" NAME NOT NULL, "c😾ol%q5_12" CHAR, col5_13 TEXT NOT NULL, "cͮol""5_14" REGROLE, col5_15 VARCHAR NOT NULL, col5_16 FLOAT4, "col5""_17" INT8 NOT NULL GENERATED ALWAYS AS ("col5 _1" + col5_4) STORED, col5_18 INT8 NOT NULL GENERATED ALWAYS AS (col5_4 + "col5 _1") STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON "t""ab'le5" ((col5_16 + 0.33156630396842957::FLOAT8) ASC, "col5 _1" DESC, col5_8 DESC, col5_16, col5_13 ASC, "c😾ol%q5_12", "%qcol5_7" DESC) INCLUDE (col5_0, col5_2, c͊ol5_5, "col5_\\x2c9", col5_15);
        postgres_1    | 	CREATE INDEX ON "t""ab'le5" ("col5 _1" ASC, "cͮol""5_14", col5_18 DESC, (col5_4 + "col5 _1") DESC, "col5""_17" DESC, (CASE WHEN "cͮol""5_14" IS NULL THEN e'X T\x12'::TEXT ELSE e'\x00'::TEXT END) DESC) INCLUDE (col5_0, col5_3, col5_4, "co"" %a1L5_\n6", "%qcol5_7", "col5_\\x2c9", "c😾ol%q5_12", col5_15);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.52s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.13s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "t""ab'le5" (col5_0 TIMESTAMP NOT NULL, "col5 _1" INT8 NOT NULL, col5_2 BYTEA NOT NULL, col5_3 JSONB NOT NULL, col5_4 INT2 NOT NULL, c͊ol5_5 TIME, "co"" %a1L5_\n6" JSONB NOT NULL, "%qcol5_7" TIME NOT NULL, col5_8 BOOL, "col5_\\x2c9" BYTEA NOT NULL, col5_10 JSONB NOT NULL, " col5_11" NAME NOT NULL, "c😾ol%q5_12" CHAR, col5_13 TEXT NOT NULL, "cͮol""5_14" REGROLE, col5_15 VARCHAR NOT NULL, col5_16 FLOAT4, "col5""_17" INT8 NOT NULL GENERATED ALWAYS AS ("col5 _1" + col5_4) STORED, col5_18 INT8 NOT NULL GENERATED ALWAYS AS (col5_4 + "col5 _1") STORED);
        test_1        |             CREATE UNIQUE INDEX ON "t""ab'le5" ((col5_16 + 0.33156630396842957::FLOAT8) ASC, "col5 _1" DESC, col5_8 DESC, col5_16, col5_13 ASC, "c😾ol%q5_12", "%qcol5_7" DESC) INCLUDE (col5_0, col5_2, c͊ol5_5, "col5_\\x2c9", col5_15);
        test_1        |             CREATE INDEX ON "t""ab'le5" ("col5 _1" ASC, "cͮol""5_14", col5_18 DESC, (col5_4 + "col5 _1") DESC, "col5""_17" DESC, (CASE WHEN "cͮol""5_14" IS NULL THEN e'X T\x12'::TEXT ELSE e'\x00'::TEXT END) DESC) INCLUDE (col5_0, col5_3, col5_4, "co"" %a1L5_\n6", "%qcol5_7", "col5_\\x2c9", "c😾ol%q5_12", col5_15);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13193892?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13193892?buildTab=artifacts#/) on release-23.1 @ [ed4bc1e693251d62f02df4ff0ca11dee36ac8f00](https://github.com/cockroachdb/cockroach/commits/ed4bc1e693251d62f02df4ff0ca11dee36ac8f00):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (61.01s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1805869489/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("01030000000100000005000000000000000000144000000000000000400000000000001440000000000000F03F0000000000000000000000000000F03F0000000000000000000000000000004000000000000014400000000000000040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(COALESCE(5.0:::FLOAT8::FLOAT8, 8.0:::FLOAT8::FLOAT8)::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS co😑l86261
        test_1        |             	)
        test_1        |             		AS tab48634
        test_1        |             ORDER BY
        test_1        |             	co😑l86261;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(COALESCE(5.0::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8)::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS co😑l86261
        test_1        |             	)
        test_1        |             		AS tab48634
        test_1        |             ORDER BY
        test_1        |             	co😑l86261;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (689.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13271687?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13271687?buildTab=artifacts#/) on release-23.1 @ [6780f67db80cc24b1d16a4e652413f667eb103a5](https://github.com/cockroachdb/cockroach/commits/6780f67db80cc24b1d16a4e652413f667eb103a5):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.00s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 68 [running]:
        test_1        | testing.tRunner.func1.2({0x22b9380, 0xc001434020})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b9380, 0xc001434020})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f3460?, {0x31fbd68?, 0xc001412030?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31bf7a0?, 0x45f3460)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc00156e600)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc0014128a0, {0xc0014128d0?, 0x3, 0xc0000702b8?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc000092968?, 0x24bd8cf?, {0xc0014128d0?, 0x20f30c0?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f30c0?, {0x24bd8cf, 0x5}, 0x3, 0x0?, {0xc001263a48, 0x3, 0xc0011f4f50?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc000296000?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269ab98?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc00136a820)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc00136a820, 0xc001412810)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (19.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13306373?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13306373?buildTab=artifacts#/) on release-23.1 @ [0e11be28dbac84e2a9b64f4ca4925d061136e0e2](https://github.com/cockroachdb/cockroach/commits/0e11be28dbac84e2a9b64f4ca4925d061136e0e2):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (121.16s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1106572164/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(1),
        test_1        |             + 	int16(2),
        test_1        |               	s"00:00:01",
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			COALESCE(int2(cbrt(7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2, 8::INT8::INT2) AS col167701,
        test_1        |             			'00:00:01'::INTERVAL::INTERVAL AS col167702
        test_1        |             	)
        test_1        |             		AS "tab%q131598"
        test_1        |             ORDER BY
        test_1        |             	col167701 NULLS FIRST, col167702 DESC NULLS LAST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			COALESCE(int2(cbrt(7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2, 8:::INT8::INT2) AS col167701,
        test_1        |             			'00:00:01':::INTERVAL::INTERVAL AS col167702
        test_1        |             	)
        test_1        |             		AS "tab%q131598"
        test_1        |             ORDER BY
        test_1        |             	col167701 NULLS FIRST, col167702 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (749.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13316120?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13316120?buildTab=artifacts#/) on release-23.1 @ [f8798447c411b142ab92be45beb4c57a6877be1d](https://github.com/cockroachdb/cockroach/commits/f8798447c411b142ab92be45beb4c57a6877be1d):


```
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-12-28 06:17:38.937 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-12-28 06:17:38.937 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-12-28 06:17:38.937 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-12-28 06:17:38.941 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-12-28 06:17:38.946 UTC [69] LOG:  database system was shut down at 2023-12-28 06:17:38 UTC
        postgres_1    | 2023-12-28 06:17:38.951 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2023-12-28 06:17:40.693 UTC [79] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2023-12-28 06:17:40.693 UTC [79] STATEMENT:  CREATE TABLE "tabl\\U000225FDe3" (col3_0 REGTYPE NOT NULL, col3_1 TIMESTAMPTZ NOT NULL, col3_2 INTERVAL NOT NULL, col3_3 DATE NOT NULL, "col3�_😠4" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN 'l_rL'::TEXT ELSE e'\x00'::TEXT END) STORED, col3_5 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN '%N^StX'::TEXT ELSE e'y\x1c||'::TEXT END) STORED, PRIMARY KEY ("col3�_😠4", col3_5, col3_1, col3_3, col3_2, col3_0));
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabl\\U000225FDe3" (col3_1 ASC);
        postgres_1    | 	CREATE INDEX ON "tabl\\U000225FDe3" (col3_0 ASC, "col3�_😠4" ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabl\\U000225FDe3" (col3_5, col3_0);
        postgres_1    | 	CREATE INDEX ON "tabl\\U000225FDe3" (col3_3, col3_5, "col3�_😠4" DESC, col3_1 ASC);
        postgres_1    | 	CREATE INDEX ON "tabl\\U000225FDe3" (col3_2 DESC, col3_3, col3_1 ASC, (CASE WHEN col3_1 IS NULL THEN ''::TEXT ELSE '+6?5'::TEXT END) ASC, (CASE WHEN col3_2 IS NULL THEN ''::TEXT ELSE 'H_'::TEXT END) ASC, "col3�_😠4" DESC);
        postgres_1    | 	CREATE INDEX ON "tabl\\U000225FDe3" (col3_0 DESC, (CASE WHEN col3_1 IS NULL THEN 'g'::TEXT ELSE e'aE\x14\x1eX\x1a'::TEXT END) ASC, col3_3 DESC);
        postgres_1    | 	CREATE INDEX ON "tabl\\U000225FDe3" (col3_2, col3_3 DESC, "col3�_😠4", col3_1 ASC, col3_5);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.37s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (1.09s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "tabl\\U000225FDe3" (col3_0 REGTYPE NOT NULL, col3_1 TIMESTAMPTZ NOT NULL, col3_2 INTERVAL NOT NULL, col3_3 DATE NOT NULL, "col3�_😠4" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN 'l_rL'::TEXT ELSE e'\x00'::TEXT END) STORED, col3_5 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN '%N^StX'::TEXT ELSE e'y\x1c||'::TEXT END) STORED, PRIMARY KEY ("col3�_😠4", col3_5, col3_1, col3_3, col3_2, col3_0));
        test_1        |             CREATE UNIQUE INDEX ON "tabl\\U000225FDe3" (col3_1 ASC);
        test_1        |             CREATE INDEX ON "tabl\\U000225FDe3" (col3_0 ASC, "col3�_😠4" ASC);
        test_1        |             CREATE UNIQUE INDEX ON "tabl\\U000225FDe3" (col3_5, col3_0);
        test_1        |             CREATE INDEX ON "tabl\\U000225FDe3" (col3_3, col3_5, "col3�_😠4" DESC, col3_1 ASC);
        test_1        |             CREATE INDEX ON "tabl\\U000225FDe3" (col3_2 DESC, col3_3, col3_1 ASC, (CASE WHEN col3_1 IS NULL THEN ''::TEXT ELSE '+6?5'::TEXT END) ASC, (CASE WHEN col3_2 IS NULL THEN ''::TEXT ELSE 'H_'::TEXT END) ASC, "col3�_😠4" DESC);
        test_1        |             CREATE INDEX ON "tabl\\U000225FDe3" (col3_0 DESC, (CASE WHEN col3_1 IS NULL THEN 'g'::TEXT ELSE e'aE\x14\x1eX\x1a'::TEXT END) ASC, col3_3 DESC);
        test_1        |             CREATE INDEX ON "tabl\\U000225FDe3" (col3_2, col3_3 DESC, "col3�_😠4", col3_1 ASC, col3_5);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13326693?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13326693?buildTab=artifacts#/) on release-23.1 @ [80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0](https://github.com/cockroachdb/cockroach/commits/80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0):


```
        test_1        |             ORDER BY
        test_1        |             	col371940 DESC NULLS LAST;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 383207 [running]:
        test_1        | testing.tRunner.func1.2({0x22b9440, 0xc000e98c60})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b9440, 0xc000e98c60})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f3460?, {0x31fbec8?, 0xc002db57d0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31bf900?, 0x45f3460)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc00118c600)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc00033afc0, {0xc00483a690?, 0x5, 0xc00033afc0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc003f86968?, 0x24bd98f?, {0xc00483a690?, 0x20f3180?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f3180?, {0x24bd98f, 0x5}, 0x5, 0x0?, {0xc002bb7a48, 0x3, 0xc00088a0f0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc00333e000?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269acf8?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc00333e000)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc00333e000, 0xc00033ad50)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (271.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13343009?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13343009?buildTab=artifacts#/) on release-23.1 @ [80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0](https://github.com/cockroachdb/cockroach/commits/80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0):


```
        cockroach1_1  | external I/O path:   /cockroach-data/extern
        cockroach1_1  | store[0]:            path=/cockroach-data
        cockroach1_1  | storage engine:      pebble
        cockroach1_1  | clusterID:           3319c1a2-445d-4e86-b38c-c69f6f8eba76
        cockroach1_1  | status:              initialized new cluster
        cockroach1_1  | nodeID:              1
        postgres_1    | CREATE EXTENSION
        postgres_1    | Loading PostGIS extensions into postgres
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | 2023-12-31 06:18:15.357 UTC [49] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2023-12-31 06:18:15.358 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2023-12-31 06:18:15.360 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2023-12-31 06:18:15.362 UTC [51] LOG:  shutting down
        postgres_1    | 2023-12-31 06:18:15.456 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2023-12-31 06:18:15.477 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2023-12-31 06:18:15.477 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2023-12-31 06:18:15.477 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2023-12-31 06:18:15.480 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2023-12-31 06:18:15.484 UTC [68] LOG:  database system was shut down at 2023-12-31 06:18:15 UTC
        postgres_1    | 2023-12-31 06:18:15.489 UTC [1] LOG:  database system is ready to accept connections
        test_1        | --- FAIL: TestCompare (605.84s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (0.30s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:172: ALTER TABLE table2 ADD FOREIGN KEY (col2_4, col2_1) REFERENCES table4 ("co|l4_2", col4_3)
        test_1        |         compare_test.go:173: cockroach1: exec: ERROR: there is no unique constraint matching given keys for referenced table table4 (SQLSTATE 23503)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.78s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13351233?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13351233?buildTab=artifacts#/) on release-23.1 @ [80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0](https://github.com/cockroachdb/cockroach/commits/80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0):


```
        postgres_1    | 		)
        postgres_1    | 			AS tab106957
        postgres_1    | 	ORDER BY
        postgres_1    | 		"co \\u0702l199028"
        postgres_1    | 2024-01-01 06:20:03.480 UTC [78] ERROR:  collation "en_us" for encoding "UTF8" does not exist at character 34
        postgres_1    | 2024-01-01 06:20:03.480 UTC [78] STATEMENT:  SELECT * FROM (SELECT NULL::TEXT COLLATE en_US AS "co l199295") AS ta😌b107077 ORDER BY "co l199295" DESC NULLS LAST
        test_1        | --- FAIL: TestCompare (737.78s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (133.76s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare484068011/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("010100000000000000008036C00000000000E05040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('G':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS "col?199308")
        test_1        |             		AS tabͭ107083
        test_1        |             ORDER BY
        test_1        |             	"col?199308" NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('G'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS "col?199308")
        test_1        |             		AS tabͭ107083
        test_1        |             ORDER BY
        test_1        |             	"col?199308" NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (761.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13359413?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13359413?buildTab=artifacts#/) on release-23.1 @ [80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0](https://github.com/cockroachdb/cockroach/commits/80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (124.67s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3841460181/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000001000000050000000000000000001C4000000000000010400000000000001C40000000000000224000000000000014400000000000002240000000000000144000000000000010400000000000001C400000000000001040"),
        test_1        |             + 	string("010300000000000000"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(7.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 5.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "Col153529"
        test_1        |             	)
        test_1        |             		AS tab91997
        test_1        |             ORDER BY
        test_1        |             	"Col153529" ASC NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(7.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 5.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "Col153529"
        test_1        |             	)
        test_1        |             		AS tab91997
        test_1        |             ORDER BY
        test_1        |             	"Col153529" ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (754.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13422239?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13422239?buildTab=artifacts#/) on release-23.1 @ [d3d457a45f468c9e7befdf5787f8a99bf46ca56e](https://github.com/cockroachdb/cockroach/commits/d3d457a45f468c9e7befdf5787f8a99bf46ca56e):


```
        postgres_1    | 			WHERE
        postgres_1    | 				NULL::BOOL
        postgres_1    | 		)
        postgres_1    | 			AS "tab*281151"
        postgres_1    | 	ORDER BY
        postgres_1    | 		col500950, " col500951", col500952 NULLS FIRST, col500953 ASC, coͧl500954 DESC, c😊ol500955 ASC NULLS FIRST
        test_1        | --- FAIL: TestCompare (970.76s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (365.42s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2856536606/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(0),
        test_1        |             + 	int16(1),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2((6.0:::FLOAT8::FLOAT8::FLOAT8 / 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "&c'ol501019")
        test_1        |             		AS tab281190
        test_1        |             ORDER BY
        test_1        |             	"&c'ol501019" NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2((6.0::FLOAT8::FLOAT8::FLOAT8 / 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS "&c'ol501019")
        test_1        |             		AS tab281190
        test_1        |             ORDER BY
        test_1        |             	"&c'ol501019" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (995.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13433041?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13433041?buildTab=artifacts#/) on release-23.1 @ [d3d457a45f468c9e7befdf5787f8a99bf46ca56e](https://github.com/cockroachdb/cockroach/commits/d3d457a45f468c9e7befdf5787f8a99bf46ca56e):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (401.46s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2820945269/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000001440000000000000204000000000000014400000000000001840000000000000104000000000000018400000000000001040000000000000204000000000000014400000000000002040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(5.0:::FLOAT8::FLOAT8::FLOAT8, 8.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col606808
        test_1        |             	)
        test_1        |             		AS tab314559
        test_1        |             ORDER BY
        test_1        |             	col606808 DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(5.0::FLOAT8::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col606808
        test_1        |             	)
        test_1        |             		AS tab314559
        test_1        |             ORDER BY
        test_1        |             	col606808 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1031.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13461008?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13461008?buildTab=artifacts#/) on release-23.1 @ [5a4e1d0bbf4b0a153b4f38ed19553d8d82aa09b7](https://github.com/cockroachdb/cockroach/commits/5a4e1d0bbf4b0a153b4f38ed19553d8d82aa09b7):


```
        postgres_1    | Loading PostGIS extensions into postgres
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2024-01-09 06:21:07.191 UTC [50] LOG:  received fast shutdown request
        postgres_1    | .2024-01-09 06:21:07.192 UTC [50] LOG:  aborting any active transactions
        postgres_1    | 2024-01-09 06:21:07.194 UTC [50] LOG:  background worker "logical replication launcher" (PID 57) exited with exit code 1
        postgres_1    | 2024-01-09 06:21:07.196 UTC [52] LOG:  shutting down
        postgres_1    | 2024-01-09 06:21:07.283 UTC [50] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-01-09 06:21:07.312 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-01-09 06:21:07.312 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-01-09 06:21:07.312 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-01-09 06:21:07.315 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-01-09 06:21:07.319 UTC [69] LOG:  database system was shut down at 2024-01-09 06:21:07 UTC
        postgres_1    | 2024-01-09 06:21:07.324 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-01-09 06:21:08.496 UTC [79] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-01-09 06:21:08.496 UTC [79] STATEMENT:  CREATE TABLE "t\\U000953DEable3" (col3_0 TIMESTAMPTZ NOT NULL, "col 3_1" REGCLASS NOT NULL, col3_2 "char" NOT NULL, col3_3 BYTEA NOT NULL, col3_4 BIT(40) NOT NULL, col3_5 TEXT NOT NULL, "c ol3_6" CHAR NOT NULL, col3_7 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col 3_1" IS NULL THEN e'u\x02Yy\x13A'::TEXT ELSE e'\x00'::TEXT END) STORED, PRIMARY KEY (col3_7, "c ol3_6", col3_4, col3_0, "col 3_1", col3_2));
        postgres_1    | 	CREATE INDEX ON "t\\U000953DEable3" (col3_5 DESC, col3_7, "c ol3_6" ASC, col3_3, "col 3_1" DESC, col3_4 DESC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.23s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.14s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "t\\U000953DEable3" (col3_0 TIMESTAMPTZ NOT NULL, "col 3_1" REGCLASS NOT NULL, col3_2 "char" NOT NULL, col3_3 BYTEA NOT NULL, col3_4 BIT(40) NOT NULL, col3_5 TEXT NOT NULL, "c ol3_6" CHAR NOT NULL, col3_7 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col 3_1" IS NULL THEN e'u\x02Yy\x13A'::TEXT ELSE e'\x00'::TEXT END) STORED, PRIMARY KEY (col3_7, "c ol3_6", col3_4, col3_0, "col 3_1", col3_2));
        test_1        |             CREATE INDEX ON "t\\U000953DEable3" (col3_5 DESC, col3_7, "c ol3_6" ASC, col3_3, "col 3_1" DESC, col3_4 DESC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13478837?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13478837?buildTab=artifacts#/) on release-23.1 @ [fab7a0da665af168257597af173d5d00197e96ba](https://github.com/cockroachdb/cockroach/commits/fab7a0da665af168257597af173d5d00197e96ba):


```
        test_1        |             - 				Exp:              -17,
        test_1        |             + 				Exp:              -14,
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               			{Int: s"9", Status: 2},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 2, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(exp(5:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || (9:::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS "c""ol316766"
        test_1        |             	)
        test_1        |             		AS tab284884
        test_1        |             ORDER BY
        test_1        |             	"c""ol316766" DESC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(exp(5::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || (9::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]::DECIMAL[])::DECIMAL[]
        test_1        |             				AS "c""ol316766"
        test_1        |             	)
        test_1        |             		AS tab284884
        test_1        |             ORDER BY
        test_1        |             	"c""ol316766" DESC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (959.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13511536?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13511536?buildTab=artifacts#/) on release-23.1 @ [afc4445618b256dd03df7beb7965c5b72d471c2a](https://github.com/cockroachdb/cockroach/commits/afc4445618b256dd03df7beb7965c5b72d471c2a):


```
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public."tablE4" AS tab120569
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab120569."c\\u242Col4_2" DESC NULLS LAST, tab120569.tableoid ASC NULLS LAST
        postgres_1    | 		)
        postgres_1    | 			AS ta🙃b120570
        postgres_1    | 	ORDER BY
        postgres_1    | 		"c😩ol|221488" ASC, co̓😘l221489 NULLS LAST, co😑l221490 NULLS LAST, col221491
        test_1        | --- FAIL: TestCompare (744.52s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (140.53s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3034727472/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("01010000000000000000E050400000000000E050C0"),
        test_1        |             + 	nil,
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('J'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS cǒl221650) AS tab120671
        test_1        |             ORDER BY
        test_1        |             	cǒl221650 ASC NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('J':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS cǒl221650) AS tab120671
        test_1        |             ORDER BY
        test_1        |             	cǒl221650 ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (768.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13590702?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13590702?buildTab=artifacts#/) on release-23.1 @ [801ed373e467acaad112eb9abad1f7d3b706efcb](https://github.com/cockroachdb/cockroach/commits/801ed373e467acaad112eb9abad1f7d3b706efcb):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (63.75s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1879977598/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	[]uint8{0x01, 0x00, 0x10, 0x06},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepoint(8.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, NULL::INT4::INT4, 2:::INT8::INT4::INT4, 7:::INT8::INT4::INT4)::BYTES
        test_1        |             				AS col95979
        test_1        |             	)
        test_1        |             		AS "%vtab53314"
        test_1        |             ORDER BY
        test_1        |             	col95979 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_astwkb(st_makepoint(8.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, NULL::INT4::INT4, 2::INT8::INT4::INT4, 7::INT8::INT4::INT4)::BYTEA
        test_1        |             				AS col95979
        test_1        |             	)
        test_1        |             		AS "%vtab53314"
        test_1        |             ORDER BY
        test_1        |             	col95979 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (692.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13629068?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13629068?buildTab=artifacts#/) on release-23.1 @ [30465dbfea8a515a5d66a766d1fccd4f30d23386](https://github.com/cockroachdb/cockroach/commits/30465dbfea8a515a5d66a766d1fccd4f30d23386):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 896608 [running]:
        test_1        | testing.tRunner.func1.2({0x22b8ce0, 0xc00c9739a0})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b8ce0, 0xc00c9739a0})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f2400?, {0x31fb8c8?, 0xc00cc7c870?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31bf1e0?, 0x45f2400)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc00c929080)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc00c9140f0, {0xc00c914120?, 0x3, 0xc00c9140f0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc00ea5f968?, 0x24bd279?, {0xc00c914120?, 0x20f29a0?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f29a0?, {0x24bd279, 0x5}, 0x3, 0x0?, {0xc00d297a48, 0x3, 0xc000fc83c0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc00a1a89c0?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269a638?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc00a1a89c0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc00a1a89c0, 0xc00c914090)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (628.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13638977?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13638977?buildTab=artifacts#/) on release-23.1 @ [15528c1be10f36f8f0ac7ca733f31bca62bb0605](https://github.com/cockroachdb/cockroach/commits/15528c1be10f36f8f0ac7ca733f31bca62bb0605):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (224.32s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1935425219/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	s"18",
        test_1        |             + 	s"19",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			char_length(('A':::STRING::STRING::STRING || sin(9.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::STRING::STRING)::INT8
        test_1        |             				AS "Col249902"
        test_1        |             	)
        test_1        |             		AS tab158237
        test_1        |             ORDER BY
        test_1        |             	"Col249902" DESC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			char_length(('A'::TEXT::TEXT::TEXT || sin(9.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::TEXT::TEXT)::INT8
        test_1        |             				AS "Col249902"
        test_1        |             	)
        test_1        |             		AS tab158237
        test_1        |             ORDER BY
        test_1        |             	"Col249902" DESC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (854.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13656950?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13656950?buildTab=artifacts#/) on release-23.1 @ [0a2b3a178fad81a0e93af53e0535f04aaecb12ce](https://github.com/cockroachdb/cockroach/commits/0a2b3a178fad81a0e93af53e0535f04aaecb12ce):


```
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2024-01-23 06:18:58.649 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2024-01-23 06:18:58.651 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2024-01-23 06:18:58.653 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2024-01-23 06:18:58.655 UTC [51] LOG:  shutting down
        test_1        | I240123 06:18:58.726684 25 testutils/soon.go:41  [-] 1  SucceedsSoon: failed to connect to ``host=postgres user=postgres database=postgres``: dial error (dial tcp 172.18.0.4:5432: connect: connection refused)
        postgres_1    | 2024-01-23 06:18:58.751 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-01-23 06:18:58.872 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-01-23 06:18:58.872 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-01-23 06:18:58.872 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-01-23 06:18:58.875 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-01-23 06:18:58.880 UTC [68] LOG:  database system was shut down at 2024-01-23 06:18:58 UTC
        postgres_1    | 2024-01-23 06:18:58.885 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-01-23 06:19:00.669 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-01-23 06:19:00.669 UTC [78] STATEMENT:  CREATE TABLE table1 (col1_0 INT8[] NULL, col1_1 BOX2D NOT NULL, col1_2 BOOL NOT NULL, col1_3 TEXT NOT NULL, col1_4 FLOAT8 NOT NULL, "col1 _5" NAME[] NOT NULL, coͭl1_6 REGTYPE NOT NULL, col1_7 CHAR NOT NULL, "col1_ 8" REGROLE NULL, "😄c%vol1_9" INT4 NOT NULL, "_\\U000F27FDcol1_10" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, "%qco�l1_\\U0003A8C211" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1 _5" IS NULL THEN e'\x100=O'::TEXT ELSE e'\x0387'::TEXT END) STORED, "Col1_12" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1 _5" IS NULL THEN e'Zw\x18V\x14y\x12T'::TEXT ELSE e'\x00'::TEXT END) STORED, col1_13 TEXT NULL GENERATED ALWAYS AS (CASE WHEN coͭl1_6 IS NULL THEN e'4\x10Pg&'::TEXT ELSE e'\x03J)&\x02J\x16'::TEXT END) STORED, PRIMARY KEY (col1_2, "😄c%vol1_9", coͭl1_6));
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 ("col1_ 8", col1_2 ASC, lower(CAST(col1_2 AS TEXT)) ASC, col1_13, coͭl1_6 DESC, col1_0, "😄c%vol1_9" ASC, col1_3, col1_7 ASC, "Col1_12" DESC, "%qco�l1_\\U0003A8C211" ASC, "col1 _5" ASC) INCLUDE (col1_4);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_13 ASC, col1_7 ASC, col1_3 ASC, "Col1_12" ASC) INCLUDE ("col1_ 8");
        postgres_1    | 	CREATE INDEX ON table1 (col1_4, "Col1_12" DESC, col1_3, "col1_ 8" ASC, col1_0 DESC, "%qco�l1_\\U0003A8C211", col1_7 ASC, "_\\U000F27FDcol1_10" ASC, col1_2 ASC, coͭl1_6 DESC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.82s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.93s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table1 (col1_0 INT8[] NULL, col1_1 BOX2D NOT NULL, col1_2 BOOL NOT NULL, col1_3 TEXT NOT NULL, col1_4 FLOAT8 NOT NULL, "col1 _5" NAME[] NOT NULL, coͭl1_6 REGTYPE NOT NULL, col1_7 CHAR NOT NULL, "col1_ 8" REGROLE NULL, "😄c%vol1_9" INT4 NOT NULL, "_\\U000F27FDcol1_10" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, "%qco�l1_\\U0003A8C211" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1 _5" IS NULL THEN e'\x100=O'::TEXT ELSE e'\x0387'::TEXT END) STORED, "Col1_12" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col1 _5" IS NULL THEN e'Zw\x18V\x14y\x12T'::TEXT ELSE e'\x00'::TEXT END) STORED, col1_13 TEXT NULL GENERATED ALWAYS AS (CASE WHEN coͭl1_6 IS NULL THEN e'4\x10Pg&'::TEXT ELSE e'\x03J)&\x02J\x16'::TEXT END) STORED, PRIMARY KEY (col1_2, "😄c%vol1_9", coͭl1_6));
        test_1        |             CREATE UNIQUE INDEX ON table1 ("col1_ 8", col1_2 ASC, lower(CAST(col1_2 AS TEXT)) ASC, col1_13, coͭl1_6 DESC, col1_0, "😄c%vol1_9" ASC, col1_3, col1_7 ASC, "Col1_12" DESC, "%qco�l1_\\U0003A8C211" ASC, "col1 _5" ASC) INCLUDE (col1_4);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_13 ASC, col1_7 ASC, col1_3 ASC, "Col1_12" ASC) INCLUDE ("col1_ 8");
        test_1        |             CREATE INDEX ON table1 (col1_4, "Col1_12" DESC, col1_3, "col1_ 8" ASC, col1_0 DESC, "%qco�l1_\\U0003A8C211", col1_7 ASC, "_\\U000F27FDcol1_10" ASC, col1_2 ASC, coͭl1_6 DESC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13713807?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13713807?buildTab=artifacts#/) on release-23.1 @ [066eece037196a6a07514de834a4233a0eb995d5](https://github.com/cockroachdb/cockroach/commits/066eece037196a6a07514de834a4233a0eb995d5):


```
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || (3:::DECIMAL::DECIMAL::DECIMAL ^ 1:::INT8::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col393347,
        test_1        |             			NULL::TIMETZ AS col393348
        test_1        |             	)
        test_1        |             		AS "%vtab290353"
        test_1        |             ORDER BY
        test_1        |             	col393347 ASC, col393348 ASC NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || (3::DECIMAL::DECIMAL::DECIMAL ^ 1::INT8::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col393347,
        test_1        |             			NULL::TIMETZ AS col393348
        test_1        |             	)
        test_1        |             		AS "%vtab290353"
        test_1        |             ORDER BY
        test_1        |             	col393347 ASC, col393348 ASC NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1033.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13723989?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13723989?buildTab=artifacts#/) on release-23.1 @ [066eece037196a6a07514de834a4233a0eb995d5](https://github.com/cockroachdb/cockroach/commits/066eece037196a6a07514de834a4233a0eb995d5):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (80.02s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1308658285/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000002040000000000000000000000000000020400000000000002240000000000000184000000000000022400000000000001840000000000000000000000000000020400000000000000000"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(8.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col83933
        test_1        |             	)
        test_1        |             		AS "t😳a̒b 65856"
        test_1        |             ORDER BY
        test_1        |             	col83933;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(8.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col83933
        test_1        |             	)
        test_1        |             		AS "t😳a̒b 65856"
        test_1        |             ORDER BY
        test_1        |             	col83933;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (709.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13731878?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13731878?buildTab=artifacts#/) on release-23.1 @ [066eece037196a6a07514de834a4233a0eb995d5](https://github.com/cockroachdb/cockroach/commits/066eece037196a6a07514de834a4233a0eb995d5):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.01s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 43 [running]:
        test_1        | testing.tRunner.func1.2({0x22b9360, 0xc00217e7c0})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b9360, 0xc00217e7c0})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f3400?, {0x31fce68?, 0xc0020897d0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:377 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31c0780?, 0x45f3400)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:310 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc0024f1380)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:227 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc002088480, {0xc002067b80?, 0x4, 0xc002088480?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:260 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc000093968?, 0x24bd8f9?, {0xc002067b80?, 0x20f3020?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f3020?, {0x24bd8f9, 0x5}, 0x4, 0x0?, {0xc0005c9a48, 0x3, 0xc001e7afa0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc0002f8000?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269acd8?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc0020071e0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc0020071e0, 0xc0020883f0)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (18.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13747394?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13747394?buildTab=artifacts#/) on release-23.1 @ [42a4b16971d1b0227f9ee9ce6fb472f37b8b8b6d](https://github.com/cockroachdb/cockroach/commits/42a4b16971d1b0227f9ee9ce6fb472f37b8b8b6d):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (18.07s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare4288660715/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(28),
        test_1        |             + 	int16(29),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			int2((sinh(6.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 / 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2
        test_1        |             				AS col25296
        test_1        |             	)
        test_1        |             		AS "%qtab
14188"
        test_1        |             ORDER BY
        test_1        |             	col25296 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			int2((sinh(6.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 / 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::INT2
        test_1        |             				AS col25296
        test_1        |             	)
        test_1        |             		AS "%qtab
14188"
        test_1        |             ORDER BY
        test_1        |             	col25296 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (647.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13776745?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13776745?buildTab=artifacts#/) on release-23.1 @ [0fea8da5fc656d9108127450c91e5801e43be25f](https://github.com/cockroachdb/cockroach/commits/0fea8da5fc656d9108127450c91e5801e43be25f):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2024-02-01 06:17:21.791 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2024-02-01 06:17:21.792 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2024-02-01 06:17:21.794 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2024-02-01 06:17:21.796 UTC [51] LOG:  shutting down
        postgres_1    | 2024-02-01 06:17:21.892 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-02-01 06:17:22.014 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-02-01 06:17:22.014 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-02-01 06:17:22.014 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-02-01 06:17:22.017 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-02-01 06:17:22.021 UTC [68] LOG:  database system was shut down at 2024-02-01 06:17:21 UTC
        postgres_1    | 2024-02-01 06:17:22.026 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-02-01 06:17:23.066 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-02-01 06:17:23.066 UTC [78] STATEMENT:  CREATE TABLE "table\\xba2" (col2_0 VARCHAR NOT NULL, "co
        postgres_1    | 	l2_1" FLOAT8 NOT NULL, "col 2_2" TSVECTOR NOT NULL, 😙col2_3 INTERVAL NOT NULL, col2_4 BIT(45) NOT NULL, col2_5 GEOGRAPHY NOT NULL, "col2 _😬6" REGCLASS NOT NULL, col2_7 UUID[], col2_8 TSVECTOR NOT NULL, col2_😞9 DECIMAL, "col 2_͐10" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN 😙col2_3 IS NULL THEN e'\x11$'::TEXT ELSE e'\x00'::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_7 IS NULL THEN e':\x05Y;Wc'::TEXT ELSE NULL END) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_8 AS TEXT))) STORED, co͌l2_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_4 AS TEXT))) STORED, "%qc""ol2_14" TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_4 AS TEXT))) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(col2_0)) STORED, " c\\x90ol2_16" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_7 IS NULL THEN 'j'::TEXT ELSE e'Pf\''::TEXT END) STORED, PRIMARY KEY ("co
        postgres_1    | 	l2_1", "col 2_͐10", col2_12, " c\\x90ol2_16", col2_11, col2_4, "col2 _😬6"));
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (604.89s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.97s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "table\\xba2" (col2_0 VARCHAR NOT NULL, "co
        test_1        |             l2_1" FLOAT8 NOT NULL, "col 2_2" TSVECTOR NOT NULL, 😙col2_3 INTERVAL NOT NULL, col2_4 BIT(45) NOT NULL, col2_5 GEOGRAPHY NOT NULL, "col2 _😬6" REGCLASS NOT NULL, col2_7 UUID[], col2_8 TSVECTOR NOT NULL, col2_😞9 DECIMAL, "col 2_͐10" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN 😙col2_3 IS NULL THEN e'\x11$'::TEXT ELSE e'\x00'::TEXT END) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_7 IS NULL THEN e':\x05Y;Wc'::TEXT ELSE NULL END) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_8 AS TEXT))) STORED, co͌l2_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_4 AS TEXT))) STORED, "%qc""ol2_14" TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_4 AS TEXT))) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(col2_0)) STORED, " c\\x90ol2_16" TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_7 IS NULL THEN 'j'::TEXT ELSE e'Pf\''::TEXT END) STORED, PRIMARY KEY ("co
        test_1        |             l2_1", "col 2_͐10", col2_12, " c\\x90ol2_16", col2_11, col2_4, "col2 _😬6"));
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (628.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13797918?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13797918?buildTab=artifacts#/) on release-23.1 @ [cc06b4975515aac5e6138fce59b1f0a723b2dc56](https://github.com/cockroachdb/cockroach/commits/cc06b4975515aac5e6138fce59b1f0a723b2dc56):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare57333100/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	nil,
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000001840000000000000184000000000000018400000000000001C40000000000000F03F0000000000001C40000000000000F03F000000000000184000000000000018400000000000001840"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_1 AS col654916,
        test_1        |             			st_makeenvelope(6.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col654917
        test_1        |             	)
        test_1        |             		AS tab397395
        test_1        |             ORDER BY
        test_1        |             	col654916 ASC, col654917 DESC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_1 AS col654916,
        test_1        |             			st_makeenvelope(6.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col654917
        test_1        |             	)
        test_1        |             		AS tab397395
        test_1        |             ORDER BY
        test_1        |             	col654916 ASC, col654917 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1143.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13818370?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13818370?buildTab=artifacts#/) on release-23.1 @ [1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638](https://github.com/cockroachdb/cockroach/commits/1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare748953906/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	nil,
        test_1        |             - 	nil,
        test_1        |             + 	[]uint8{0x61, 0x08, 0x06, 0xa0, 0x1f, 0xc0, 0x3e, 0x04},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_2 AS col393066,
        test_1        |             			st_astwkb(st_makepointm(2.0:::FLOAT8::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 3:::INT8::INT4::INT4, 9:::INT8::INT4::INT4, NULL::INT4::INT4)::BYTES
        test_1        |             				AS col393067
        test_1        |             	)
        test_1        |             		AS "t
ab360356"
        test_1        |             ORDER BY
        test_1        |             	col393066 DESC NULLS LAST, col393067;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			NULL::rand_typ_2 AS col393066,
        test_1        |             			st_astwkb(st_makepointm(2.0::FLOAT8::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 3::INT8::INT4::INT4, 9::INT8::INT4::INT4, NULL::INT4::INT4)::BYTEA
        test_1        |             				AS col393067
        test_1        |             	)
        test_1        |             		AS "t
ab360356"
        test_1        |             ORDER BY
        test_1        |             	col393066 DESC NULLS LAST, col393067;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1189.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13830537?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13830537?buildTab=artifacts#/) on release-23.1 @ [1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638](https://github.com/cockroachdb/cockroach/commits/1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638):


```
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-02-04 06:16:56.715 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-02-04 06:16:56.715 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-02-04 06:16:56.715 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-02-04 06:16:56.718 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-02-04 06:16:56.723 UTC [68] LOG:  database system was shut down at 2024-02-04 06:16:56 UTC
        postgres_1    | 2024-02-04 06:16:56.727 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-02-04 06:16:58.460 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-02-04 06:16:58.460 UTC [78] STATEMENT:  CREATE TABLE table2 ("col2&_0" BYTEA NOT NULL, "col2_%a01" TIME, "co l2\\uF2A6_2" INT4 NOT NULL, col2_3 TSQUERY NOT NULL, "col2_'4" DECIMAL NOT NULL, "col2*_5" INTERVAL NOT NULL, "col2_�6" UUID NOT NULL, col2_7 REGROLE NULL, "col%v2_8" VARCHAR, "c
ol2_9" DATE NOT NULL, "col 2_10" TEXT NOT NULL, col2_11 TIMETZ NOT NULL, "co!l2_12" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_13 TEXT GENERATED ALWAYS AS (lower(CAST("col2_%a01" AS TEXT))) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2*_5" IS NULL THEN e'\x05 6<'::TEXT ELSE e'M\\Q\x07lN|'::TEXT END) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (lower("col 2_10")) STORED);
        postgres_1    | 	CREATE INDEX ON table2 (col2_14 DESC, (CASE WHEN "col2&_0" IS NULL THEN e'\x02&-N6".'::TEXT ELSE e'lY|+\x1a7'::TEXT END) ASC, "co!l2_12" ASC, col2_7, "col2_%a01" DESC, "col2&_0" ASC, (CASE WHEN "col2*_5" IS NULL THEN e'\x00'::TEXT ELSE '``}I%'::TEXT END)) INCLUDE (col2_3, "col2*_5", "c
ol2_9");
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_11, (CASE WHEN col2_7 IS NULL THEN e'sZ\r/'::TEXT ELSE '"'::TEXT END) DESC, "col2_�6" ASC, "col2_'4" DESC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_14 ASC, col2_13 ASC, "col 2_10" DESC, col2_7 ASC, "c
ol2_9" DESC, "co l2\\uF2A6_2" DESC) INCLUDE ("col2_'4");
        postgres_1    | 	CREATE INDEX ON table2 ("col2_�6" DESC, col2_13, "co l2\\uF2A6_2" DESC, col2_7, "col 2_10" ASC, "co!l2_12", "col2_'4", "c
ol2_9" DESC, lower(CAST(col2_3 AS TEXT)) ASC, col2_14 ASC, "col2&_0", "col2_%a01" ASC, col2_11, "col2*_5", col2_15 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_7 DESC);
        postgres_1    | 	CREATE INDEX ON table2 (lower(CAST("col2_�6" AS TEXT)) DESC, "c
ol2_9", "col2_'4" ASC, "col2_%a01") INCLUDE (col2_3, "col2*_5", "col%v2_8", "col 2_10", col2_11, col2_13);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_7 ASC, lower(CAST("col2_�6" AS TEXT)), col2_11 ASC, col2_13, "c
ol2_9" DESC, "col2_%a01" DESC) INCLUDE ("col2&_0", "col2_'4", "col%v2_8");
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.56s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (0.89s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE table2 ("col2&_0" BYTEA NOT NULL, "col2_%a01" TIME, "co l2\\uF2A6_2" INT4 NOT NULL, col2_3 TSQUERY NOT NULL, "col2_'4" DECIMAL NOT NULL, "col2*_5" INTERVAL NOT NULL, "col2_�6" UUID NOT NULL, col2_7 REGROLE NULL, "col%v2_8" VARCHAR, "c
ol2_9" DATE NOT NULL, "col 2_10" TEXT NOT NULL, col2_11 TIMETZ NOT NULL, "co!l2_12" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_13 TEXT GENERATED ALWAYS AS (lower(CAST("col2_%a01" AS TEXT))) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "col2*_5" IS NULL THEN e'\x05 6<'::TEXT ELSE e'M\\Q\x07lN|'::TEXT END) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (lower("col 2_10")) STORED);
        test_1        |             CREATE INDEX ON table2 (col2_14 DESC, (CASE WHEN "col2&_0" IS NULL THEN e'\x02&-N6".'::TEXT ELSE e'lY|+\x1a7'::TEXT END) ASC, "co!l2_12" ASC, col2_7, "col2_%a01" DESC, "col2&_0" ASC, (CASE WHEN "col2*_5" IS NULL THEN e'\x00'::TEXT ELSE '``}I%'::TEXT END)) INCLUDE (col2_3, "col2*_5", "c
ol2_9");
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_11, (CASE WHEN col2_7 IS NULL THEN e'sZ\r/'::TEXT ELSE '"'::TEXT END) DESC, "col2_�6" ASC, "col2_'4" DESC);
        test_1        |             CREATE INDEX ON table2 (col2_14 ASC, col2_13 ASC, "col 2_10" DESC, col2_7 ASC, "c
ol2_9" DESC, "co l2\\uF2A6_2" DESC) INCLUDE ("col2_'4");
        test_1        |             CREATE INDEX ON table2 ("col2_�6" DESC, col2_13, "co l2\\uF2A6_2" DESC, col2_7, "col 2_10" ASC, "co!l2_12", "col2_'4", "c
ol2_9" DESC, lower(CAST(col2_3 AS TEXT)) ASC, col2_14 ASC, "col2&_0", "col2_%a01" ASC, col2_11, "col2*_5", col2_15 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_7 DESC);
        test_1        |             CREATE INDEX ON table2 (lower(CAST("col2_�6" AS TEXT)) DESC, "c
ol2_9", "col2_'4" ASC, "col2_%a01") INCLUDE (col2_3, "col2*_5", "col%v2_8", "col 2_10", col2_11, col2_13);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_7 ASC, lower(CAST("col2_�6" AS TEXT)), col2_11 ASC, col2_13, "c
ol2_9" DESC, "col2_%a01" DESC) INCLUDE ("col2&_0", "col2_'4", "col%v2_8");
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13840946?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13840946?buildTab=artifacts#/) on release-23.1 @ [1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638](https://github.com/cockroachdb/cockroach/commits/1ba29d87c4ffe1e8f9c2670f6e31187dc9c9f638):


```
        cockroach1_1  | clusterID:           ebd494fd-193c-40dc-ab2f-11a88025819d
        cockroach1_1  | status:              initialized new cluster
        cockroach1_1  | nodeID:              1
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2024-02-05 06:17:23.804 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2024-02-05 06:17:23.806 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2024-02-05 06:17:23.807 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2024-02-05 06:17:23.810 UTC [51] LOG:  shutting down
        test_1        | I240205 06:17:23.904268 8 testutils/soon.go:41  [-] 1  SucceedsSoon: failed to connect to ``host=postgres user=postgres database=postgres``: dial error (dial tcp 172.18.0.3:5432: connect: connection refused)
        postgres_1    | 2024-02-05 06:17:23.906 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-02-05 06:17:24.027 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-02-05 06:17:24.028 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-02-05 06:17:24.028 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-02-05 06:17:24.031 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-02-05 06:17:24.035 UTC [68] LOG:  database system was shut down at 2024-02-05 06:17:23 UTC
        postgres_1    | 2024-02-05 06:17:24.040 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-02-05 06:17:26.304 UTC [78] ERROR:  foreign key constraint "ta ble4_col4_7_col4_\\U000107FF6_fkey" cannot be implemented
        postgres_1    | 2024-02-05 06:17:26.304 UTC [78] DETAIL:  Key columns "col4_7" and "col1_%q3" are of incompatible types: name and "char".
        postgres_1    | 2024-02-05 06:17:26.304 UTC [78] STATEMENT:  ALTER TABLE "ta ble4" ADD FOREIGN KEY (col4_7, "col4_\\U000107FF6") REFERENCES table1 ("col1_%q3", col1_1) ON DELETE RESTRICT
        test_1        | --- FAIL: TestCompare (606.49s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.39s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: ALTER TABLE "ta ble4" ADD FOREIGN KEY (col4_7, "col4_\\U000107FF6") REFERENCES table1 ("col1_%q3", col1_1) ON DELETE RESTRICT
        test_1        |         compare_test.go:173: postgres: exec: ERROR: foreign key constraint "ta ble4_col4_7_col4_\\U000107FF6_fkey" cannot be implemented (SQLSTATE 42804)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #117263 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.14-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112510 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.12-rc.FROZEN]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13919413?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13919413?buildTab=artifacts#/) on release-23.1 @ [cdcfef58fdc6f3aba98d17aa794775653318c837](https://github.com/cockroachdb/cockroach/commits/cdcfef58fdc6f3aba98d17aa794775653318c837):


```
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (160.36s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare308141376/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000001C40709A74761DA54C400000000000001C400000000000000040000000000000104000000000000000400000000000001040709A74761DA54C400000000000001C40709A74761DA54C40"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(7.0:::FLOAT8::FLOAT8::FLOAT8, cotd(1.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 4.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col191014
        test_1        |             	)
        test_1        |             		AS tab127020
        test_1        |             ORDER BY
        test_1        |             	col191014 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(7.0::FLOAT8::FLOAT8::FLOAT8, cotd(1.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 4.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col191014
        test_1        |             	)
        test_1        |             		AS tab127020
        test_1        |             ORDER BY
        test_1        |             	col191014 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (789.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13935494?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13935494?buildTab=artifacts#/) on release-23.1 @ [9b3c1cabbcbfb491d33b6ea8943c0762a17892b6](https://github.com/cockroachdb/cockroach/commits/9b3c1cabbcbfb491d33b6ea8943c0762a17892b6):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (486.41s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1388053721/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("0103000000010000000500000000000000000008400000000000001C40000000000000084000000000000018400000000000000040000000000000184000000000000000400000000000001C4000000000000008400000000000001C40"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0:::FLOAT8::FLOAT8::FLOAT8, 7.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "co\\uC45Bl 481379"
        test_1        |             	)
        test_1        |             		AS tab279646
        test_1        |             ORDER BY
        test_1        |             	"co\\uC45Bl 481379";
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(3.0::FLOAT8::FLOAT8::FLOAT8, 7.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "co\\uC45Bl 481379"
        test_1        |             	)
        test_1        |             		AS tab279646
        test_1        |             ORDER BY
        test_1        |             	"co\\uC45Bl 481379";
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1116.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13990686?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/13990686?buildTab=artifacts#/) on release-23.1 @ [78de5d3123c3ee76fb2eaf37bb3f11f29b74c331](https://github.com/cockroachdb/cockroach/commits/78de5d3123c3ee76fb2eaf37bb3f11f29b74c331):


```
        postgres_1    | 				"	ta͎b43534"."c%vol1_15" ASC NULLS LAST, "	ta͎b43534".col1_6 NULLS FIRST
        postgres_1    | 		)
        postgres_1    | 			AS tab43537
        postgres_1    | 	ORDER BY
        postgres_1    | 		"col.76391" NULLS LAST, "c
        postgres_1    | 	ol76392" ASC, " coL76393" DESC NULLS LAST, "co l76394" NULLS FIRST
        test_1        | --- FAIL: TestCompare (672.94s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (68.02s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3684196784/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("01010000000000000000205CC00000000000E05040"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('C':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS "c ol&76424")
        test_1        |             		AS t̨ab̙43564
        test_1        |             ORDER BY
        test_1        |             	"c ol&76424" NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('C'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS "c ol&76424")
        test_1        |             		AS t̨ab̙43564
        test_1        |             ORDER BY
        test_1        |             	"c ol&76424" NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (697.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14006098?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14006098?buildTab=artifacts#/) on release-23.1 @ [f75e5557799190dfa4afd13bf9f4f227495a6331](https://github.com/cockroachdb/cockroach/commits/f75e5557799190dfa4afd13bf9f4f227495a6331):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2833765000/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("0H"),
        test_1        |             + 	string("0.0000000000000000H"),
        test_1        |               	[16]uint8{0x00, 0x00, 0x00, 0x00, ...},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((0:::DECIMAL::DECIMAL::DECIMAL ^ 5:::INT8::INT8::INT8)::DECIMAL::DECIMAL || 'H':::STRING::STRING::STRING)::STRING
        test_1        |             				AS " col665151",
        test_1        |             			'00000000-0000-0000-0000-000000000006':::UUID::UUID AS col665152
        test_1        |             	)
        test_1        |             		AS "t%qab338396"
        test_1        |             ORDER BY
        test_1        |             	" col665151" DESC NULLS LAST, col665152 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((0::DECIMAL::DECIMAL::DECIMAL ^ 5::INT8::INT8::INT8)::DECIMAL::DECIMAL || 'H'::TEXT::TEXT::TEXT)::TEXT
        test_1        |             				AS " col665151",
        test_1        |             			'00000000-0000-0000-0000-000000000006'::UUID::UUID AS col665152
        test_1        |             	)
        test_1        |             		AS "t%qab338396"
        test_1        |             ORDER BY
        test_1        |             	" col665151" DESC NULLS LAST, col665152 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1168.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14023217?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14023217?buildTab=artifacts#/) on release-23.1 @ [0c48ab97d8935fa4360a70803c80c1cef71fd958](https://github.com/cockroachdb/cockroach/commits/0c48ab97d8935fa4360a70803c80c1cef71fd958):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (0.01s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 36 [running]:
        test_1        | testing.tRunner.func1.2({0x22b9e40, 0xc00071f8e0})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b9e40, 0xc00071f8e0})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f5400?, {0x31fe508?, 0xc0025ff6b0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:375 +0x436
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31c1e20?, 0x45f5400)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:308 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc00237fc80)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:225 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc002055860, {0xc002204280?, 0x5, 0xc002055860?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:258 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc00221d968?, 0x24be6e9?, {0xc002204280?, 0x20f39c0?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:96 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f39c0?, {0x24be6e9, 0x5}, 0x5, 0x0?, {0xc000e89a48, 0x3, 0xc00221e000?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc000296000?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269bb78?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc000757ba0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc000757ba0, 0xc0020557d0)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (18.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14039100?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14039100?buildTab=artifacts#/) on release-23.1 @ [0c48ab97d8935fa4360a70803c80c1cef71fd958](https://github.com/cockroachdb/cockroach/commits/0c48ab97d8935fa4360a70803c80c1cef71fd958):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (56.23s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare2945367623/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("7.3890560989306502272I"),
        test_1        |             + 	string("7.3890560989306502I"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(exp(2::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'I'::TEXT::TEXT::TEXT)::TEXT
        test_1        |             				AS "co l69789"
        test_1        |             	)
        test_1        |             		AS tab37658
        test_1        |             ORDER BY
        test_1        |             	"co l69789" DESC;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(exp(2:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || 'I':::STRING::STRING::STRING)::STRING
        test_1        |             				AS "co l69789"
        test_1        |             	)
        test_1        |             		AS tab37658
        test_1        |             ORDER BY
        test_1        |             	"co l69789" DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (685.81s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14059828?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14059828?buildTab=artifacts#/) on release-23.1 @ [0c48ab97d8935fa4360a70803c80c1cef71fd958](https://github.com/cockroachdb/cockroach/commits/0c48ab97d8935fa4360a70803c80c1cef71fd958):


```
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 2, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((5:::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]::DECIMAL[] || log(4:::DECIMAL::DECIMAL::DECIMAL, 4:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS "%7ecol333683",
        test_1        |             			NULL::DECIMAL AS col333684
        test_1        |             	)
        test_1        |             		AS tab182231
        test_1        |             ORDER BY
        test_1        |             	"%7ecol333683" DESC, col333684 ASC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((5::DECIMAL::DECIMAL::DECIMAL || NULL::DECIMAL[]::DECIMAL[])::DECIMAL[]::DECIMAL[] || log(4::DECIMAL::DECIMAL::DECIMAL, 4::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS "%7ecol333683",
        test_1        |             			NULL::DECIMAL AS col333684
        test_1        |             	)
        test_1        |             		AS tab182231
        test_1        |             ORDER BY
        test_1        |             	"%7ecol333683" DESC, col333684 ASC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (862.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14090158?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14090158?buildTab=artifacts#/) on release-23.1 @ [333253387f07fcbfc20697df5556224d44ca86c2](https://github.com/cockroachdb/cockroach/commits/333253387f07fcbfc20697df5556224d44ca86c2):


```
        test_1        |         compare_test.go:172: CREATE TABLE table4 ("col4_̞\\u16F50" INTERVAL NOT NULL, col4_1 INT2 NOT NULL, col😊4_2 TIMESTAMP NOT NULL, col4̞_3 FLOAT4, col4_4 TSQUERY NOT NULL, col4_5 NAME, "col4_'6" OID, col_4_7 REGCLASS NULL, col4_8 TIMESTAMPTZ, col4_9 INT8 NOT NULL, col4_10 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col_4_7 IS NULL THEN e'\x00'::TEXT ELSE '"'::TEXT END) STORED, col4_11 INT8 NOT NULL GENERATED ALWAYS AS (col4_9 + col4_1) STORED, "\\U000482A2col4_12" INT8 NOT NULL GENERATED ALWAYS AS (col4_9 + col4_1) STORED, col4_13 TEXT GENERATED ALWAYS AS (lower(CAST("col4_'6" AS TEXT))) STORED, col4_14 INT8 NOT NULL GENERATED ALWAYS AS (col4_9 + col4_1) STORED, col4😺_😂15 TEXT GENERATED ALWAYS AS (CASE WHEN col4_8 IS NULL THEN e'?"\x10'::TEXT ELSE NULL END) STORED);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:141: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 52 [running]:
        test_1        | testing.tRunner.func1.2({0x22b9e40, 0xc0015132e0})
        test_1        | 	GOROOT/src/testing/testing.go:1396 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1399 +0x39f
        test_1        | panic({0x22b9e40, 0xc0015132e0})
        test_1        | 	GOROOT/src/runtime/panic.go:884 +0x212
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x45f5400?, {0x31fe508?, 0xc000f276b0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:375 +0x436
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x31c1e20?, 0x45f5400)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:308 +0x13c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc000ec1080)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:225 +0x96
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc0005357a0, {0xc000b67ac0?, 0x4, 0xc0005357a0?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:258 +0x71f
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc000093968?, 0x24be6e9?, {0xc000b67ac0?, 0x20f39c0?, 0x1?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:96 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x20f39c0?, {0x24be6e9, 0x5}, 0x4, 0x0?, {0xc000d91a48, 0x3, 0xc000057270?})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:93 +0x16a
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x1?, 0x60a9c0?, {0x0, 0x0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:104 +0x194
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc000515380?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:84 +0x37
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x269bb78?)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:51 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc000515380)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:143 +0xed
        test_1        | testing.tRunner(0xc000515380, 0xc000535740)
        test_1        | 	GOROOT/src/testing/testing.go:1446 +0x10b
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1493 +0x35f
        compare_test_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 2
--- FAIL: TestComposeCompare (22.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14142028?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14142028?buildTab=artifacts#/) on release-23.1 @ [1ae85d2986e88578216a11776be952f63e593e87](https://github.com/cockroachdb/cockroach/commits/1ae85d2986e88578216a11776be952f63e593e87):


```
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3802305229/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	int32(8),
        test_1        |             - 	nil,
        test_1        |             + 	string("01010000000000000000E050C00000000000803640"),
        test_1        |               	string("0011000000000000000000000000000000"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			8:::INT8::INT4 AS col87494,
        test_1        |             			st_pointfromgeohash('D':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS " col87495",
        test_1        |             			B'0011'::BIT(34) AS col87496
        test_1        |             	)
        test_1        |             		AS tͤab55983
        test_1        |             ORDER BY
        test_1        |             	col87494 DESC NULLS LAST, " col87495", col87496 ASC;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			8::INT8::INT4 AS col87494,
        test_1        |             			st_pointfromgeohash('D'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS " col87495",
        test_1        |             			B'0011'::BIT(34) AS col87496
        test_1        |             	)
        test_1        |             		AS tͤab55983
        test_1        |             ORDER BY
        test_1        |             	col87494 DESC NULLS LAST, " col87495", col87496 ASC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (699.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14148232?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14148232?buildTab=artifacts#/) on release-23.1 @ [1ae85d2986e88578216a11776be952f63e593e87](https://github.com/cockroachdb/cockroach/commits/1ae85d2986e88578216a11776be952f63e593e87):


```
        postgres_1    | 					AS col180170,
        postgres_1    | 				NULL::VOID AS "co""�l180171"
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public."ta\\u4E6Cble1" AS tab105217
        postgres_1    | 		)
        postgres_1    | 			AS tab105218
        postgres_1    | 	ORDER BY
        postgres_1    | 		col180170 DESC NULLS FIRST, "co""�l180171" ASC NULLS LAST
        test_1        | --- FAIL: TestCompare (732.31s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (128.43s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare520597402/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010100000000000000008036C00000000000803640"),
        test_1        |             + 	nil,
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('E'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS col180189) AS "tab'105233"
        test_1        |             ORDER BY
        test_1        |             	col180189 NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_pointfromgeohash('E':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS col180189) AS "tab'105233"
        test_1        |             ORDER BY
        test_1        |             	col180189 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (756.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14238634?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14238634?buildTab=artifacts#/) on release-23.1 @ [5e06334602f87f353dd62884f1db78e0678cd167](https://github.com/cockroachdb/cockroach/commits/5e06334602f87f353dd62884f1db78e0678cd167):


```
        postgres_1    | 		" col534113" NULLS FIRST,
        postgres_1    | 		"%qco\\u3FA7l534114" NULLS FIRST
        postgres_1    | 2024-03-05 06:25:40.110 UTC [78] ERROR:  collation "da_dk" for encoding "UTF8" does not exist at character 34
        postgres_1    | 2024-03-05 06:25:40.110 UTC [78] STATEMENT:  SELECT * FROM (SELECT NULL::TEXT COLLATE da_DK AS col534157) AS "\\u0649tab358908" ORDER BY col534157 DESC NULLS LAST
        postgres_1    | 2024-03-05 06:25:40.224 UTC [78] ERROR:  collation "en_us" for encoding "UTF8" does not exist at character 34
        postgres_1    | 2024-03-05 06:25:40.224 UTC [78] STATEMENT:  SELECT * FROM (SELECT NULL::TEXT COLLATE en_US AS "col �534281") AS tab359000 ORDER BY "col �534281" ASC NULLS LAST
        test_1        | --- FAIL: TestCompare (1086.35s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (481.20s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare120400919/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	int16(2),
        test_1        |             + 	int16(3),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(acosh(COALESCE(NULL::FLOAT8, 9.0:::FLOAT8::FLOAT8)::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS col534329)
        test_1        |             		AS "tab\\uC17E359041"
        test_1        |             ORDER BY
        test_1        |             	col534329 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT int2(acosh(COALESCE(NULL::FLOAT8, 9.0::FLOAT8::FLOAT8)::FLOAT8)::FLOAT8::FLOAT8)::INT2 AS col534329)
        test_1        |             		AS "tab\\uC17E359041"
        test_1        |             ORDER BY
        test_1        |             	col534329 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1111.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14316637?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14316637?buildTab=artifacts#/) on release-23.1 @ [e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe](https://github.com/cockroachdb/cockroach/commits/e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe):


```
        postgres_1    | 		"%vcol205245" ASC NULLS FIRST,
        postgres_1    | 		"ĉol%p205246" DESC NULLS LAST,
        postgres_1    | 		col205247 DESC NULLS LAST,
        postgres_1    | 		co͜l205248 DESC NULLS FIRST,
        postgres_1    | 		col205249 ASC NULLS FIRST,
        postgres_1    | 		col205250 ASC NULLS FIRST
        test_1        | --- FAIL: TestCompare (753.97s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (149.55s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3566802910/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("E0.84509804001425683071"),
        test_1        |             + 	string("E0.8450980400142568"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT ('E'::TEXT::TEXT::TEXT || log(7::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::TEXT AS col205273)
        test_1        |             		AS "😘tab%112304"
        test_1        |             ORDER BY
        test_1        |             	col205273 DESC;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT ('E':::STRING::STRING::STRING || log(7:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::STRING AS col205273)
        test_1        |             		AS "😘tab%112304"
        test_1        |             ORDER BY
        test_1        |             	col205273 DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (778.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14327157?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14327157?buildTab=artifacts#/) on release-23.1 @ [e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe](https://github.com/cockroachdb/cockroach/commits/e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (292.47s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare835955247/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000001000000050000000000000000000040000000000000204000000000000000400000000000000000000000000000224000000000000000000000000000002240000000000000204000000000000000400000000000002040"),
        test_1        |             + 	string("010300000000000000"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(2.0:::FLOAT8::FLOAT8::FLOAT8, 8.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8, 0.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col353867
        test_1        |             	)
        test_1        |             		AS tab183415
        test_1        |             ORDER BY
        test_1        |             	col353867 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(2.0::FLOAT8::FLOAT8::FLOAT8, 8.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8, 0.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col353867
        test_1        |             	)
        test_1        |             		AS tab183415
        test_1        |             ORDER BY
        test_1        |             	col353867 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (922.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14347073?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14347073?buildTab=artifacts#/) on release-23.1 @ [e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe](https://github.com/cockroachdb/cockroach/commits/e0fe00bd7e976fd9f33c097bd29b1fb2a64268fe):


```
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-03-12 05:18:06.237 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-03-12 05:18:06.237 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-03-12 05:18:06.237 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-03-12 05:18:06.240 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-03-12 05:18:06.245 UTC [68] LOG:  database system was shut down at 2024-03-12 05:18:06 UTC
        postgres_1    | 2024-03-12 05:18:06.250 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-03-12 05:18:07.936 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-03-12 05:18:07.936 UTC [78] STATEMENT:  CREATE TABLE "ta'ble1" (col1_0 BIT(32), col1_1 BOOL NOT NULL, "col.1_2" BYTEA NULL, "col%v1_3" REGCLASS NULL, col1_4 BIT(10), col1_5 INT2, "col 1_6" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, "cOl1_7" INT8 GENERATED ALWAYS AS (col1_5 + (-24324)::INT8) STORED, "\\xdec o%eel1,_8" TEXT GENERATED ALWAYS AS (lower(CAST(col1_4 AS TEXT))) STORED, col1_9 TEXT GENERATED ALWAYS AS (lower(CAST(col1_0 AS TEXT))) STORED, "co\\u1BB9l1_10" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" ASC, "\\xdec o%eel1,_8" ASC, col1_5 DESC, "col 1_6", "cOl1_7") INCLUDE (col1_1, "col.1_2", col1_9);
        postgres_1    | 	CREATE UNIQUE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" DESC, "cOl1_7", col1_0 ASC, "col%v1_3" ASC, col1_9 DESC, col1_1, "\\xdec o%eel1,_8", lower(CAST(col1_4 AS TEXT)) DESC, lower(CAST(col1_1 AS TEXT))) INCLUDE (col1_4, col1_5);
        postgres_1    | 	CREATE INDEX ON "ta'ble1" (lower(CAST(col1_4 AS TEXT)) ASC, col1_9 ASC, col1_1, "cOl1_7", "col%v1_3" DESC) INCLUDE (col1_0, col1_5);
        postgres_1    | 	CREATE INDEX ON "ta'ble1" ("col 1_6" ASC, col1_5, col1_4, "co\\u1BB9l1_10", (CASE WHEN "col.1_2" IS NULL THEN e'hMP\x03Ke\x16'::TEXT ELSE ''::TEXT END) ASC, col1_9 DESC, (CASE WHEN "col%v1_3" IS NULL THEN e'\x00'::TEXT ELSE '``'::TEXT END) ASC, col1_0, col1_1, "col%v1_3" ASC, "cOl1_7" DESC);
        postgres_1    | 	CREATE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" DESC, "cOl1_7", col1_4, col1_9, "col 1_6" ASC, col1_0 ASC, col1_5 DESC, col1_1 DESC, (CASE WHEN "col%v1_3" IS NULL THEN e'L\x1cy\t\x16I'::TEXT ELSE '3%dgKZc'::TEXT END) DESC, "col.1_2") INCLUDE ("col%v1_3");
        postgres_1    | 	CREATE INDEX ON "ta'ble1" ("col.1_2" DESC, col1_4, "col%v1_3" DESC, col1_5 ASC);
        postgres_1    | 	CREATE INDEX ON "ta'ble1" (col1_9 ASC, col1_1, col1_5, "col%v1_3", "cOl1_7", col1_4 DESC) INCLUDE (col1_0);
        postgres_1    | 	CREATE INDEX ON "ta'ble1" (col1_4 ASC, col1_5, col1_1 ASC, "col 1_6" DESC, col1_0, (CASE WHEN "col%v1_3" IS NULL THEN ']'::TEXT ELSE e',,! /B\r'::TEXT END), lower(CAST(col1_0 AS TEXT)) DESC, "col.1_2" DESC, col1_9, "\\xdec o%eel1,_8", "co\\u1BB9l1_10" ASC);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.72s)
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (0.89s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "ta'ble1" (col1_0 BIT(32), col1_1 BOOL NOT NULL, "col.1_2" BYTEA NULL, "col%v1_3" REGCLASS NULL, col1_4 BIT(10), col1_5 INT2, "col 1_6" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, "cOl1_7" INT8 GENERATED ALWAYS AS (col1_5 + (-24324)::INT8) STORED, "\\xdec o%eel1,_8" TEXT GENERATED ALWAYS AS (lower(CAST(col1_4 AS TEXT))) STORED, col1_9 TEXT GENERATED ALWAYS AS (lower(CAST(col1_0 AS TEXT))) STORED, "co\\u1BB9l1_10" TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED);
        test_1        |             CREATE UNIQUE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" ASC, "\\xdec o%eel1,_8" ASC, col1_5 DESC, "col 1_6", "cOl1_7") INCLUDE (col1_1, "col.1_2", col1_9);
        test_1        |             CREATE UNIQUE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" DESC, "cOl1_7", col1_0 ASC, "col%v1_3" ASC, col1_9 DESC, col1_1, "\\xdec o%eel1,_8", lower(CAST(col1_4 AS TEXT)) DESC, lower(CAST(col1_1 AS TEXT))) INCLUDE (col1_4, col1_5);
        test_1        |             CREATE INDEX ON "ta'ble1" (lower(CAST(col1_4 AS TEXT)) ASC, col1_9 ASC, col1_1, "cOl1_7", "col%v1_3" DESC) INCLUDE (col1_0, col1_5);
        test_1        |             CREATE INDEX ON "ta'ble1" ("col 1_6" ASC, col1_5, col1_4, "co\\u1BB9l1_10", (CASE WHEN "col.1_2" IS NULL THEN e'hMP\x03Ke\x16'::TEXT ELSE ''::TEXT END) ASC, col1_9 DESC, (CASE WHEN "col%v1_3" IS NULL THEN e'\x00'::TEXT ELSE '``'::TEXT END) ASC, col1_0, col1_1, "col%v1_3" ASC, "cOl1_7" DESC);
        test_1        |             CREATE INDEX ON "ta'ble1" ("co\\u1BB9l1_10" DESC, "cOl1_7", col1_4, col1_9, "col 1_6" ASC, col1_0 ASC, col1_5 DESC, col1_1 DESC, (CASE WHEN "col%v1_3" IS NULL THEN e'L\x1cy\t\x16I'::TEXT ELSE '3%dgKZc'::TEXT END) DESC, "col.1_2") INCLUDE ("col%v1_3");
        test_1        |             CREATE INDEX ON "ta'ble1" ("col.1_2" DESC, col1_4, "col%v1_3" DESC, col1_5 ASC);
        test_1        |             CREATE INDEX ON "ta'ble1" (col1_9 ASC, col1_1, col1_5, "col%v1_3", "cOl1_7", col1_4 DESC) INCLUDE (col1_0);
        test_1        |             CREATE INDEX ON "ta'ble1" (col1_4 ASC, col1_5, col1_1 ASC, "col 1_6" DESC, col1_0, (CASE WHEN "col%v1_3" IS NULL THEN ']'::TEXT ELSE e',,! /B\r'::TEXT END), lower(CAST(col1_0 AS TEXT)) DESC, "col.1_2" DESC, col1_9, "\\xdec o%eel1,_8", "co\\u1BB9l1_10" ASC);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14366930?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14366930?buildTab=artifacts#/) on release-23.1 @ [d91b11dc5fb32e6c7e8402f1e03001bcae01faa4](https://github.com/cockroachdb/cockroach/commits/d91b11dc5fb32e6c7e8402f1e03001bcae01faa4):


```
        postgres_1    | 2024-03-13 05:17:47.920 UTC [68] LOG:  database system was shut down at 2024-03-13 05:17:47 UTC
        postgres_1    | 2024-03-13 05:17:47.925 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-03-13 05:27:51.510 UTC [98] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-03-13 05:27:51.510 UTC [98] STATEMENT:  CREATE TABLE "tabLe3" (col3_0 INT8 NOT NULL, col3_1 BOX2D, col3_2 INT4 NULL, "co""l3_3" VARCHAR NULL, col3_4 REGROLE NULL, col3_5 FLOAT8 NOT NULL, " %vcol3_6" REGTYPE NULL, col3_7 TEXT NOT NULL, "col	3_8" OID NULL, col3_9 TIMETZ NOT NULL, "c�ol3_10" TIMESTAMPTZ NOT NULL, "/col 3_11" REGCLASS, col3_12 INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED, "col
        postgres_1    | 	3_13" TEXT NULL GENERATED ALWAYS AS (lower("co""l3_3")) STORED, "col*""3_14" INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED, col3_15 INT8 NOT NULL GENERATED ALWAYS AS (col3_0 + (-5145320716356222592)::INT8) STORED, col3_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "c�ol3_10" IS NULL THEN ''::TEXT ELSE e'|\x01G\x18rW'::TEXT END) STORED, col3̩_17 INT8 NULL GENERATED ALWAYS AS (col3_0 + col3_2) STORED, col3_18 INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED);
        postgres_1    | 	CREATE INDEX ON "tabLe3" (col3̩_17 ASC, " %vcol3_6" ASC, "col
        postgres_1    | 	3_13" ASC, "co""l3_3" ASC, col3_2, col3_5, col3_12 ASC) INCLUDE (col3_0, col3_1, col3_7, "col	3_8", col3_9, "/col 3_11", col3_16);
        postgres_1    | 	CREATE INDEX ON "tabLe3" (col3_15, "/col 3_11" DESC, (CASE WHEN "c�ol3_10" IS NULL THEN '"'::TEXT ELSE e'[\x024ii'::TEXT END) ASC, col3_2, col3_16 DESC, col3̩_17 ASC, "col	3_8", col3_7 DESC, (col3_0 + col3_2) DESC, col3_5, "c�ol3_10", col3_4 ASC, col3_18);
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabLe3" (col3̩_17, col3_4, "c�ol3_10" ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabLe3" ((col3_5 + (-0.0)::FLOAT8) ASC, lower("co""l3_3") ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabLe3" (col3_12, col3_7, " %vcol3_6", col3_2 ASC, (col3_0 + col3_2) ASC, col3_5 DESC, "/col 3_11" DESC, "c�ol3_10" DESC, col3̩_17, "co""l3_3" DESC, lower(CAST("col	3_8" AS TEXT)) ASC, "col
        postgres_1    | 	3_13", col3_4, lower("co""l3_3"), col3_16 DESC) INCLUDE (col3_1, "col	3_8", col3_9);
        postgres_1    | 	CREATE INDEX ON "tabLe3" (col3_9 DESC, col3_12 DESC, col3̩_17 DESC, col3_15 DESC, col3_7, col3_5 ASC, (CASE WHEN " %vcol3_6" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) DESC, col3_18 ASC, col3_0 ASC, "col	3_8" ASC, "co""l3_3" DESC) INCLUDE (col3_2, col3_4, "c�ol3_10", "col
        postgres_1    | 	3_13");
        postgres_1    | 	CREATE UNIQUE INDEX ON "tabLe3" (col3_16 ASC, "col	3_8" DESC, " %vcol3_6" ASC, col3_9 DESC) INCLUDE (col3_0, col3_1, col3_2, "co""l3_3", col3_12, col3_18);
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.83s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.74s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "tabLe3" (col3_0 INT8 NOT NULL, col3_1 BOX2D, col3_2 INT4 NULL, "co""l3_3" VARCHAR NULL, col3_4 REGROLE NULL, col3_5 FLOAT8 NOT NULL, " %vcol3_6" REGTYPE NULL, col3_7 TEXT NOT NULL, "col	3_8" OID NULL, col3_9 TIMETZ NOT NULL, "c�ol3_10" TIMESTAMPTZ NOT NULL, "/col 3_11" REGCLASS, col3_12 INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED, "col
        test_1        |             3_13" TEXT NULL GENERATED ALWAYS AS (lower("co""l3_3")) STORED, "col*""3_14" INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED, col3_15 INT8 NOT NULL GENERATED ALWAYS AS (col3_0 + (-5145320716356222592)::INT8) STORED, col3_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN "c�ol3_10" IS NULL THEN ''::TEXT ELSE e'|\x01G\x18rW'::TEXT END) STORED, col3̩_17 INT8 NULL GENERATED ALWAYS AS (col3_0 + col3_2) STORED, col3_18 INT8 NULL GENERATED ALWAYS AS (col3_2 + col3_0) STORED);
        test_1        |             CREATE INDEX ON "tabLe3" (col3̩_17 ASC, " %vcol3_6" ASC, "col
        test_1        |             3_13" ASC, "co""l3_3" ASC, col3_2, col3_5, col3_12 ASC) INCLUDE (col3_0, col3_1, col3_7, "col	3_8", col3_9, "/col 3_11", col3_16);
        test_1        |             CREATE INDEX ON "tabLe3" (col3_15, "/col 3_11" DESC, (CASE WHEN "c�ol3_10" IS NULL THEN '"'::TEXT ELSE e'[\x024ii'::TEXT END) ASC, col3_2, col3_16 DESC, col3̩_17 ASC, "col	3_8", col3_7 DESC, (col3_0 + col3_2) DESC, col3_5, "c�ol3_10", col3_4 ASC, col3_18);
        test_1        |             CREATE UNIQUE INDEX ON "tabLe3" (col3̩_17, col3_4, "c�ol3_10" ASC);
        test_1        |             CREATE UNIQUE INDEX ON "tabLe3" ((col3_5 + (-0.0)::FLOAT8) ASC, lower("co""l3_3") ASC);
        test_1        |             CREATE UNIQUE INDEX ON "tabLe3" (col3_12, col3_7, " %vcol3_6", col3_2 ASC, (col3_0 + col3_2) ASC, col3_5 DESC, "/col 3_11" DESC, "c�ol3_10" DESC, col3̩_17, "co""l3_3" DESC, lower(CAST("col	3_8" AS TEXT)) ASC, "col
        test_1        |             3_13", col3_4, lower("co""l3_3"), col3_16 DESC) INCLUDE (col3_1, "col	3_8", col3_9);
        test_1        |             CREATE INDEX ON "tabLe3" (col3_9 DESC, col3_12 DESC, col3̩_17 DESC, col3_15 DESC, col3_7, col3_5 ASC, (CASE WHEN " %vcol3_6" IS NULL THEN e'\x00'::TEXT ELSE ''::TEXT END) DESC, col3_18 ASC, col3_0 ASC, "col	3_8" ASC, "co""l3_3" DESC) INCLUDE (col3_2, col3_4, "c�ol3_10", "col
        test_1        |             3_13");
        test_1        |             CREATE UNIQUE INDEX ON "tabLe3" (col3_16 ASC, "col	3_8" DESC, " %vcol3_6" ASC, col3_9 DESC) INCLUDE (col3_0, col3_1, col3_2, "co""l3_3", col3_12, col3_18);
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (630.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14385507?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14385507?buildTab=artifacts#/) on release-23.1 @ [220bde1772035a94d92521b86d9ad07f118abfb8](https://github.com/cockroachdb/cockroach/commits/220bde1772035a94d92521b86d9ad07f118abfb8):


```
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1417167074/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare postgres to cockroach1: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |               	strings.Join({
        test_1        |               		"01010000400000000000004C40000000000000F03FE",
        test_1        |             - 		"4",
        test_1        |             + 		"3",
        test_1        |               		"A1498FE7321BC0",
        test_1        |               	}, ""),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makepointm((8.0::FLOAT8::FLOAT8::FLOAT8 * 7.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8, tan(8.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "�col😙615332"
        test_1        |             	)
        test_1        |             		AS "t%vab300962"
        test_1        |             ORDER BY
        test_1        |             	"�col😙615332" DESC;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makepointm((8.0:::FLOAT8::FLOAT8::FLOAT8 * 7.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8, tan(8.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS "�col😙615332"
        test_1        |             	)
        test_1        |             		AS "t%vab300962"
        test_1        |             ORDER BY
        test_1        |             	"�col😙615332" DESC;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (1064.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14429204?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14429204?buildTab=artifacts#/) on release-23.1 @ [21b39df19e06186269f5af0c9fdd35da1404f046](https://github.com/cockroachdb/cockroach/commits/21b39df19e06186269f5af0c9fdd35da1404f046):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (51.13s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare3404696928/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	s"0",
        test_1        |             + 	s"5",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_npoints(st_makeenvelope(7.0:::FLOAT8::FLOAT8::FLOAT8, 2.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::INT8
        test_1        |             				AS col51365
        test_1        |             	)
        test_1        |             		AS "	t ab25754"
        test_1        |             ORDER BY
        test_1        |             	col51365 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_npoints(st_makeenvelope(7.0::FLOAT8::FLOAT8::FLOAT8, 2.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::INT8
        test_1        |             				AS col51365
        test_1        |             	)
        test_1        |             		AS "	t ab25754"
        test_1        |             ORDER BY
        test_1        |             	col51365 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (681.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14438116?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14438116?buildTab=artifacts#/) on release-23.1 @ [21b39df19e06186269f5af0c9fdd35da1404f046](https://github.com/cockroachdb/cockroach/commits/21b39df19e06186269f5af0c9fdd35da1404f046):


```
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (36.83s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1699624035/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(0 -90,45 -45)"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_combinebbox(NULL::BOX2D::BOX2D, st_geomfromgeohash('H':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY::GEOMETRY)::BOX2D
        test_1        |             				AS col36192
        test_1        |             	)
        test_1        |             		AS 🙁tab22810
        test_1        |             ORDER BY
        test_1        |             	CAST(col36192 AS STRING);
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_combinebbox(NULL::BOX2D::BOX2D, st_geomfromgeohash('H'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY::GEOMETRY)::BOX2D
        test_1        |             				AS col36192
        test_1        |             	)
        test_1        |             		AS 🙁tab22810
        test_1        |             ORDER BY
        test_1        |             	CAST(col36192 AS TEXT);
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (666.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14453115?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14453115?buildTab=artifacts#/) on release-23.1 @ [70f8a738d1e0d0bdbe51cfdd100a01ad217d30bb](https://github.com/cockroachdb/cockroach/commits/70f8a738d1e0d0bdbe51cfdd100a01ad217d30bb):


```
        postgres_1    | 		col266786 NULLS LAST,
        postgres_1    | 		"cOl266787" ASC NULLS FIRST,
        postgres_1    | 		"coL266788" NULLS FIRST,
        postgres_1    | 		"col\\U0000CF54266789" ASC,
        postgres_1    | 		col266790 ASC NULLS FIRST,
        postgres_1    | 		"c{ol266791" ASC,
        postgres_1    | 		"'col266792",
        postgres_1    | 		col266793 ASC
        test_1        | --- FAIL: TestCompare (823.39s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (219.19s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare884432761/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	nil,
        test_1        |             + 	string("0103000000010000000500000000000000008046C0000000000080464000000000008046C00000000000805640000000000000000000000000008056400000000000000000000000000080464000000000008046C00000000000804640"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('G':::STRING::STRING::STRING, NULL::INT4::INT4)::GEOMETRY AS colͭ266881) AS t̝ab179509
        test_1        |             ORDER BY
        test_1        |             	colͭ266881 NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_geomfromgeohash('G'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::GEOMETRY AS colͭ266881) AS t̝ab179509
        test_1        |             ORDER BY
        test_1        |             	colͭ266881 NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (847.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14467106?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14467106?buildTab=artifacts#/) on release-23.1 @ [723e07a6e5e69b41fda5fe2a36693c52f4c376d3](https://github.com/cockroachdb/cockroach/commits/723e07a6e5e69b41fda5fe2a36693c52f4c376d3):


```
        test_1        |             + 				Int:              s"12163594037787468",
        test_1        |             - 				Exp:              -19,
        test_1        |             + 				Exp:              -16,
        test_1        |               				Status:           2,
        test_1        |               				NaN:              false,
        test_1        |               				InfinityModifier: s"none",
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: {{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || (9:::INT8::INT8::INT8 % ln(7:::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col112587
        test_1        |             	)
        test_1        |             		AS "tab�69141"
        test_1        |             ORDER BY
        test_1        |             	col112587 DESC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(NULL::DECIMAL[]::DECIMAL[] || (9::INT8::INT8::INT8 % ln(7::DECIMAL::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL[]
        test_1        |             				AS col112587
        test_1        |             	)
        test_1        |             		AS "tab�69141"
        test_1        |             ORDER BY
        test_1        |             	col112587 DESC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (718.24s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14483498?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14483498?buildTab=artifacts#/) on release-23.1 @ [6790290f0e6ff613fd57c405e6d13b9befb2766f](https://github.com/cockroachdb/cockroach/commits/6790290f0e6ff613fd57c405e6d13b9befb2766f):


```
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2024-03-21 05:17:09.619 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2024-03-21 05:17:09.619 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2024-03-21 05:17:09.619 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2024-03-21 05:17:09.622 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2024-03-21 05:17:09.626 UTC [68] LOG:  database system was shut down at 2024-03-21 05:17:09 UTC
        postgres_1    | 2024-03-21 05:17:09.630 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2024-03-21 05:17:10.796 UTC [78] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2024-03-21 05:17:10.796 UTC [78] STATEMENT:  CREATE TABLE "%qtable1" ("col1_""0" INTERVAL NOT NULL, "col1_ �1" INTERVAL NOT NULL, col1_2 INT2 NOT NULL, col1_3 REGCLASS, "col1""_4" INT4 NOT NULL, "c'ol1_5" REGROLE NULL, col1_6 TEXT NOT NULL, col1_7 TIMESTAMP[] NOT NULL, col1_8 "char" NOT NULL, col1_9 INET NOT NULL, col1_10 TEXT NOT NULL, col1_11 FLOAT4 NOT NULL, col1_12 TSVECTOR, "%3fcol1_13" INT8 NOT NULL GENERATED ALWAYS AS ("col1""_4" + col1_2) STORED, "col1_\n14" TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_8)) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_10 AS TEXT))) STORED, col1_16 INT8 NOT NULL GENERATED ALWAYS AS (col1_2 + "col1""_4") STORED, col1_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, " col1_18" INT8 NOT NULL GENERATED ALWAYS AS (col1_2 + "col1""_4") STORED);
        postgres_1    | 	CREATE INDEX ON "%qtable1" ("c'ol1_5", "col1_""0" DESC, col1_6 DESC, col1_8, col1_3 ASC, col1_11 DESC, col1_16, "%3fcol1_13" ASC, col1_10, "col1_\n14", col1_2 ASC, " col1_18" DESC, col1_15 DESC, col1_7 ASC) INCLUDE (col1_17);
        postgres_1    | 	CREATE INDEX ON "%qtable1" (col1_15 ASC, "col1_""0" DESC, " col1_18", col1_7, "col1_\n14" DESC, col1_8 DESC, col1_2, "col1_ �1" ASC);
        postgres_1    | 	CREATE INDEX ON "%qtable1" (col1_2 DESC, col1_15 DESC, col1_8 ASC, lower(col1_6) ASC, col1_7, "c'ol1_5") INCLUDE (col1_6, col1_10, col1_12, col1_17, " col1_18");
        postgres_1    | 	CREATE UNIQUE INDEX ON "%qtable1" (col1_11 ASC, (CASE WHEN "c'ol1_5" IS NULL THEN '55('::TEXT ELSE e'\x00'::TEXT END) DESC, "c'ol1_5", col1_2, col1_17 DESC, "col1_\n14" DESC, col1_7 ASC, col1_10 ASC, col1_16, col1_3 ASC, (CASE WHEN col1_3 IS NULL THEN e'\U00002603'::TEXT ELSE 'nerf'::TEXT END), col1_8 DESC) INCLUDE (col1_6, col1_9, "%3fcol1_13");
        postgres_1    | 	CREATE INDEX ON "%qtable1" (col1_8 ASC, col1_10, "col1_\n14", " col1_18", "col1""_4" DESC) INCLUDE (col1_2, col1_3, "c'ol1_5", col1_6, "%3fcol1_13", col1_17);
        postgres_1    | 	CREATE INDEX ON "%qtable1" ("%3fcol1_13" ASC, "col1""_4" ASC, "col1_ �1" DESC, " col1_18" ASC, "col1_""0" DESC);
        postgres_1    | 	CREATE INDEX ON "%qtable1" (col1_16 DESC, col1_3 ASC, col1_9, col1_10 DESC, col1_17 DESC, col1_8, ("col1""_4" + col1_2) DESC, "col1_ �1", "col1_""0") INCLUDE (col1_2, col1_6, col1_12, "%3fcol1_13");
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.53s)
        test_1        |     compare_test.go:132: Checking connection to: postgres
        test_1        |     compare_test.go:132: Checking connection to: cockroach1
        test_1        |     compare_test.go:132: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.02s)
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:172: CREATE TABLE "%qtable1" ("col1_""0" INTERVAL NOT NULL, "col1_ �1" INTERVAL NOT NULL, col1_2 INT2 NOT NULL, col1_3 REGCLASS, "col1""_4" INT4 NOT NULL, "c'ol1_5" REGROLE NULL, col1_6 TEXT NOT NULL, col1_7 TIMESTAMP[] NOT NULL, col1_8 "char" NOT NULL, col1_9 INET NOT NULL, col1_10 TEXT NOT NULL, col1_11 FLOAT4 NOT NULL, col1_12 TSVECTOR, "%3fcol1_13" INT8 NOT NULL GENERATED ALWAYS AS ("col1""_4" + col1_2) STORED, "col1_\n14" TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_8)) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_10 AS TEXT))) STORED, col1_16 INT8 NOT NULL GENERATED ALWAYS AS (col1_2 + "col1""_4") STORED, col1_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, " col1_18" INT8 NOT NULL GENERATED ALWAYS AS (col1_2 + "col1""_4") STORED);
        test_1        |             CREATE INDEX ON "%qtable1" ("c'ol1_5", "col1_""0" DESC, col1_6 DESC, col1_8, col1_3 ASC, col1_11 DESC, col1_16, "%3fcol1_13" ASC, col1_10, "col1_\n14", col1_2 ASC, " col1_18" DESC, col1_15 DESC, col1_7 ASC) INCLUDE (col1_17);
        test_1        |             CREATE INDEX ON "%qtable1" (col1_15 ASC, "col1_""0" DESC, " col1_18", col1_7, "col1_\n14" DESC, col1_8 DESC, col1_2, "col1_ �1" ASC);
        test_1        |             CREATE INDEX ON "%qtable1" (col1_2 DESC, col1_15 DESC, col1_8 ASC, lower(col1_6) ASC, col1_7, "c'ol1_5") INCLUDE (col1_6, col1_10, col1_12, col1_17, " col1_18");
        test_1        |             CREATE UNIQUE INDEX ON "%qtable1" (col1_11 ASC, (CASE WHEN "c'ol1_5" IS NULL THEN '55('::TEXT ELSE e'\x00'::TEXT END) DESC, "c'ol1_5", col1_2, col1_17 DESC, "col1_\n14" DESC, col1_7 ASC, col1_10 ASC, col1_16, col1_3 ASC, (CASE WHEN col1_3 IS NULL THEN e'\U00002603'::TEXT ELSE 'nerf'::TEXT END), col1_8 DESC) INCLUDE (col1_6, col1_9, "%3fcol1_13");
        test_1        |             CREATE INDEX ON "%qtable1" (col1_8 ASC, col1_10, "col1_\n14", " col1_18", "col1""_4" DESC) INCLUDE (col1_2, col1_3, "c'ol1_5", col1_6, "%3fcol1_13", col1_17);
        test_1        |             CREATE INDEX ON "%qtable1" ("%3fcol1_13" ASC, "col1""_4" ASC, "col1_ �1" DESC, " col1_18" ASC, "col1_""0" DESC);
        test_1        |             CREATE INDEX ON "%qtable1" (col1_16 DESC, col1_3 ASC, col1_9, col1_10 DESC, col1_17 DESC, col1_8, ("col1""_4" + col1_2) DESC, "col1_ �1", "col1_""0") INCLUDE (col1_2, col1_6, col1_12, "%3fcol1_13");
        test_1        |             
        test_1        |         compare_test.go:173: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (629.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #118616 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.1-rc]
- #118615 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.15-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #116006 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115907 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #115745 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.0-rc]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #109341 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14532444?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/14532444?buildTab=artifacts#/) on release-23.1 @ [6790290f0e6ff613fd57c405e6d13b9befb2766f](https://github.com/cockroachdb/cockroach/commits/6790290f0e6ff613fd57c405e6d13b9befb2766f):


```
        test_1        |         compare_test.go:141: starting test: postgres
        test_1        |         compare_test.go:150: initializing connection: cockroach1
        test_1        |         compare_test.go:150: initializing connection: postgres
        test_1        |         compare_test.go:201: open /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/018812dfcf0ce383ac3e799915a61396/TestComposeCompare1539451633/002/postgres.log: no such file or directory
        test_1        |         compare_test.go:203: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []any(Inverse(func1, []any{
        test_1        |             - 	string("010300000000000000"),
        test_1        |             + 	string("010300000001000000050000000000000000001440000000000000224000000000000014400000000000000840000000000000184000000000000008400000000000001840000000000000224000000000000014400000000000002240"),
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(5.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8, 6.0:::FLOAT8::FLOAT8::FLOAT8, 3.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col653892,
        test_1        |             			NULL::GEOGRAPHY AS "c""ol653893"
        test_1        |             	)
        test_1        |             		AS tab345861
        test_1        |             ORDER BY
        test_1        |             	col653892 ASC NULLS LAST, "c""ol653893" ASC NULLS LAST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_makeenvelope(5.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8, 6.0::FLOAT8::FLOAT8::FLOAT8, 3.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY
        test_1        |             				AS col653892,
        test_1        |             			NULL::GEOGRAPHY AS "c""ol653893"
        test_1        |             	)
        test_1        |             		AS tab345861
        test_1        |             ORDER BY
        test_1        |             	col653892 ASC NULLS LAST, "c""ol653893" ASC NULLS LAST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:124: exit status 1
--- FAIL: TestComposeCompare (986.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #120925 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-custom-v2]
- #120924 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-custom]
- #120869 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.4-rc]
- #120868 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.18-rc]
- #119903 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.1.17-rc]
- #119902 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2.3-rc]
- #116150 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-master]
- #112154 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-23.2]
- #89361 compose: TestComposeCompare failed [O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/compose/compare/compare/compare_test.go
+++ b/pkg/compose/compare/compare/compare_test.go
@@ -49,18 +49,6 @@ func TestCompare(t *testing.T) {
 		addr string
 		init []string
 	}{
-		"postgres": {
-			addr: "postgresql://postgres@postgres:5432/postgres",
-			init: []string{
-				"drop schema if exists public cascade",
-				"create schema public",
-				"CREATE EXTENSION IF NOT EXISTS postgis",
-				"CREATE EXTENSION IF NOT EXISTS postgis_topology",
-				"CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;",
-				"CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\";",
-				"CREATE EXTENSION IF NOT EXISTS pg_trgm;",
-			},
-		},
 		"cockroach1": {
 			addr: "postgresql://root@cockroach1:26257/postgres?sslmode=disable",
 			init: []string{
@@ -85,22 +73,6 @@ func TestCompare(t *testing.T) {
 		},
 	}
 	configs := map[string]testConfig{
-		"postgres": {
-			setup:           sqlsmith.Setups[sqlsmith.RandTableSetupName],
-			setupMutators:   []randgen.Mutator{randgen.PostgresCreateTableMutator},
-			opts:            []sqlsmith.SmitherOption{sqlsmith.PostgresMode()},
-			ignoreSQLErrors: true,
-			conns: []testConn{
-				{
-					name:     "cockroach1",
-					mutators: []randgen.Mutator{},
-				},
-				{
-					name:     "postgres",
-					mutators: []randgen.Mutator{randgen.PostgresMutator},
-				},
-			},
-		},
 		"mutators": {
 			setup:           sqlsmith.Setups[sqlsmith.RandTableSetupName],
 			opts:            []sqlsmith.SmitherOption{sqlsmith.CompareMode()},

--- a/pkg/compose/compare/docker-compose.yml
+++ b/pkg/compose/compare/docker-compose.yml
@@ -1,10 +1,5 @@
 version: '3'
 services:
-  postgres:
-    image: postgis/postgis:13-3.1
-    environment:
-      - POSTGRES_INITDB_ARGS=--locale=C --encoding=UTF8
-      - POSTGRES_HOST_AUTH_METHOD=trust
   cockroach1:
     image: ubuntu:xenial-20170214
     command: /cockroach/cockroach start-single-node --insecure --listen-addr cockroach1
@@ -22,7 +17,6 @@ services:
     # compare.test is a binary built by the pkg/compose/prepare.sh in non-bazel builds
     command: /compare/compare.test -each ${EACH} -test.run ${TESTS} -artifacts ${ARTIFACTS}
     depends_on:
-      - postgres
       - cockroach1
       - cockroach2
     volumes:

